
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Hands-on training course for microbial genomics and metagenomics in clinical and public health applications">
      
      
        <meta name="author" content="CIDRI-Africa">
      
      
        <link rel="canonical" href="https://cidri-africa.github.io/microbial-genomics-training/day8/metagenomics/">
      
      
      
      
      <link rel="icon" href="../../assets/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.19">
    
    
      
        <title>Metagenomics in Clinical & Public Health - Microbial Genomics & Metagenomics Training Course</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.7e37652d.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      
  


  
  

<script id="__analytics">function __md_analytics(){function e(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],e("js",new Date),e("config","G-XXXXXXXXXX"),document.addEventListener("DOMContentLoaded",(function(){document.forms.search&&document.forms.search.query.addEventListener("blur",(function(){this.value&&e("event","search",{search_term:this.value})}));document$.subscribe((function(){var t=document.forms.feedback;if(void 0!==t)for(var a of t.querySelectorAll("[type=submit]"))a.addEventListener("click",(function(a){a.preventDefault();var n=document.location.pathname,d=this.getAttribute("data-md-value");e("event","feedback",{page:n,data:d}),t.firstElementChild.disabled=!0;var r=t.querySelector(".md-feedback__note [data-md-value='"+d+"']");r&&(r.hidden=!1)})),t.hidden=!1})),location$.subscribe((function(t){e("config","G-XXXXXXXXXX",{page_path:t.pathname})}))}));var t=document.createElement("script");t.async=!0,t.src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX",document.getElementById("__analytics").insertAdjacentElement("afterEnd",t)}</script>
  
    <script>"undefined"!=typeof __md_analytics&&__md_analytics()</script>
  

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="amber">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#metagenomics-in-clinical-public-health" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Microbial Genomics &amp; Metagenomics Training Course" class="md-header__button md-logo" aria-label="Microbial Genomics & Metagenomics Training Course" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 2h2v2c0 1.44.68 2.61 1.88 3.78.86.83 2.01 1.63 3.21 2.42l-1.83 1.19C8.27 10.72 7.31 10 6.5 9.21 5.07 7.82 4 6.1 4 4zm14 0h2v2c0 2.1-1.07 3.82-2.5 5.21-1.41 1.38-3.21 2.52-4.96 3.63-1.75 1.12-3.45 2.21-4.66 3.38C6.68 17.39 6 18.56 6 20v2H4v-2c0-2.1 1.07-3.82 2.5-5.21 1.41-1.38 3.21-2.52 4.96-3.63 1.75-1.12 3.45-2.21 4.66-3.38C17.32 6.61 18 5.44 18 4zm-3.26 10.61c.99.67 1.95 1.39 2.76 2.18C18.93 16.18 20 17.9 20 20v2h-2v-2c0-1.44-.68-2.61-1.88-3.78-.86-.83-2.01-1.63-3.21-2.42zM7 3h10v1l-.06.5H7.06L7 4zm.68 3h8.64c-.24.34-.52.69-.9 1.06l-.51.44H9.07l-.49-.44c-.38-.37-.66-.72-.9-1.06m1.41 10.5h5.84l.49.44c.38.37.66.72.9 1.06H7.68c.24-.34.52-.69.9-1.06zm-2.03 3h9.88l.06.5v1H7v-1z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Microbial Genomics & Metagenomics Training Course
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Metagenomics in Clinical &amp; Public Health
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="amber"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="amber"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/CIDRI-Africa/microbial-genomics-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    CIDRI-Africa/microbial-genomics-training
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../course/overview/" class="md-tabs__link">
          
  
  
    
  
  Course Information

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../modules/day1/" class="md-tabs__link">
          
  
  
    
  
  Daily Modules

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../datasets/" class="md-tabs__link">
          
  
  
    
  
  Resources

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Microbial Genomics &amp; Metagenomics Training Course" class="md-nav__button md-logo" aria-label="Microbial Genomics & Metagenomics Training Course" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 2h2v2c0 1.44.68 2.61 1.88 3.78.86.83 2.01 1.63 3.21 2.42l-1.83 1.19C8.27 10.72 7.31 10 6.5 9.21 5.07 7.82 4 6.1 4 4zm14 0h2v2c0 2.1-1.07 3.82-2.5 5.21-1.41 1.38-3.21 2.52-4.96 3.63-1.75 1.12-3.45 2.21-4.66 3.38C6.68 17.39 6 18.56 6 20v2H4v-2c0-2.1 1.07-3.82 2.5-5.21 1.41-1.38 3.21-2.52 4.96-3.63 1.75-1.12 3.45-2.21 4.66-3.38C17.32 6.61 18 5.44 18 4zm-3.26 10.61c.99.67 1.95 1.39 2.76 2.18C18.93 16.18 20 17.9 20 20v2h-2v-2c0-1.44-.68-2.61-1.88-3.78-.86-.83-2.01-1.63-3.21-2.42zM7 3h10v1l-.06.5H7.06L7 4zm.68 3h8.64c-.24.34-.52.69-.9 1.06l-.51.44H9.07l-.49-.44c-.38-.37-.66-.72-.9-1.06m1.41 10.5h5.84l.49.44c.38.37.66.72.9 1.06H7.68c.24-.34.52-.69.9-1.06zm-2.03 3h9.88l.06.5v1H7v-1z"/></svg>

    </a>
    Microbial Genomics & Metagenomics Training Course
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/CIDRI-Africa/microbial-genomics-training" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    CIDRI-Africa/microbial-genomics-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
  
    <a href="../../course/overview/" class="md-nav__link">
      
  
  
  <span class="md-ellipsis">
    Course Information
    
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
  
    <a href="../../modules/day1/" class="md-nav__link">
      
  
  
  <span class="md-ellipsis">
    Daily Modules
    
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
    
    
    
      
      
        
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
  
    <a href="../../datasets/" class="md-nav__link">
      
  
  
  <span class="md-ellipsis">
    Resources
    
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/CIDRI-Africa/microbial-genomics-training/edit/master/docs/day8/metagenomics.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4zm10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1z"/></svg>
    </a>
  
  
    
      
    
    <a href="https://github.com/CIDRI-Africa/microbial-genomics-training/raw/master/docs/day8/metagenomics.md" title="View source of this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.2 8.2 0 0 1-1.23-2"/></svg>
    </a>
  


<h1 id="metagenomics-in-clinical-public-health">Metagenomics in Clinical &amp; Public Health<a class="headerlink" href="#metagenomics-in-clinical-public-health" title="Permanent link">&para;</a></h1>
<h2 id="1-overview">1. Overview<a class="headerlink" href="#1-overview" title="Permanent link">&para;</a></h2>
<p>This section introduces <strong>core principles of metagenomics</strong>. Metagenomics is the study of genetic material recovered directly from environmental or clinical samples, allowing 
analysis of <strong>entire microbial communities</strong> without the need for cultivation. This section includes conceptual notes, rationale for each step, practical commands, 
and a <strong>mini toy dataset exercise</strong>. Unlike genomics (WGS) which focuses on analyzing individual genomes (such as bacterial isolate), metagenomics studies the collective genomes or markers from microbial communities.</p>
<h3 id="genomics-vs-metagenomics">Genomics vs Metagenomics<a class="headerlink" href="#genomics-vs-metagenomics" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Description</th>
<th>Whole Genome Sequencing (WGS)</th>
<th>Metagenomics</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Umbrella term</strong></td>
<td>Microbial Genomics</td>
<td>Microbial Profiling</td>
</tr>
<tr>
<td><strong>Scope</strong></td>
<td>Sequencing of the entire genome of a single isolate (pure culture)</td>
<td>Sequences all genetic material in mixed community (without culturing)</td>
</tr>
<tr>
<td><strong>Applications</strong></td>
<td>Species identification, genome assembly, annotation, AMR/virulence detection, plasmid/MGE analysis, outbreak tracking</td>
<td>Community profiling, functional potential, pathogen detection in mixed samples, ecology studies</td>
</tr>
<tr>
<td><strong>Features</strong></td>
<td>Organism is known \&amp; isolated before sequencing</td>
<td>Captures both known and unknown microbes from environment or host</td>
</tr>
</tbody>
</table>
<h3 id="metagenomic-strategies-shotgun-vs-16s-rrna">Metagenomic Strategies: Shotgun vs 16S rRNA<a class="headerlink" href="#metagenomic-strategies-shotgun-vs-16s-rrna" title="Permanent link">&para;</a></h3>
<table>
<thead>
<tr>
<th>Description</th>
<th>Shotgun Metagenomics</th>
<th>16S rRNA Amplicon Gene Sequencing</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Definition</strong></td>
<td>Random sequencing of <strong>All DNA</strong> in a sample</td>
<td><strong>Targeted amplicon sequencing</strong> of the 16 S rRNA gene</td>
</tr>
<tr>
<td><strong>Resolution</strong></td>
<td>Species- and strain-level, <strong>functional genes</strong> (AMR, metabolism, plasmids, MGEs)</td>
<td>Genus-level (sometimes species-level), <strong>no direct functional</strong> information</td>
</tr>
<tr>
<td><strong>Merits</strong></td>
<td>Comprehensive (taxonomy + function), <strong>detects viruses and fungi</strong></td>
<td>Limited to taxonomic resolution, can't detect fungi/viruses, lacks functional insights, <strong>good for bacterial</strong> surveys</td>
</tr>
<tr>
<td><strong>Demerits</strong></td>
<td>More expensive, higher computational load</td>
<td>Cost-effective, standardized</td>
</tr>
</tbody>
</table>
<hr />
<h2 id="2-workflow-from-sequencing-to-interpretation">2. Workflow: From Sequencing to Interpretation<a class="headerlink" href="#2-workflow-from-sequencing-to-interpretation" title="Permanent link">&para;</a></h2>
<h3 id="step-1-study-design-sequencing">Step 1. Study Design &amp; Sequencing<a class="headerlink" href="#step-1-study-design-sequencing" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: Sequencing depth, read length, and platform choice directly influence resolution of taxa, detection of low-abundance organisms, and assembly quality.</li>
<li><strong>Example</strong>:</li>
<li>Illumina (short reads): accurate, cost-effective, widely used for clinical metagenomics.</li>
<li>ONT/PacBio (long reads): useful for resolving repeats, plasmids, MGEs.</li>
</ul>
<p>For this training we will use the data in <code>/data/users/user29/metagenomes/shotgun/</code> for shotgun metagenomics. It is important to note that, one can download shotgun metagenome sequences from NCBI-SRA using <code>ncbi-tools</code>. Install <code>ncbi-tools</code> and run</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="c1">## Fetch the data from NCBI-SRA</span>
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="c1"># fasterq-dump SRR13827118 --progress --threads 8</span>
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a><span class="c1">## Compress the files</span>
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>gzip<span class="w"> </span>SRR13827118*.fastq
</span></code></pre></div>
<hr />
<h2 id="shotgun-metagenomics">Shotgun metagenomics<a class="headerlink" href="#shotgun-metagenomics" title="Permanent link">&para;</a></h2>
<h3 id="step-2-quality-control">Step 2. Quality Control<a class="headerlink" href="#step-2-quality-control" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: To reduce the effects of sequencing errors, remove adapters, and low-quality 
reads which may reduce false positives.</li>
<li><strong>Tools</strong>: <code>fastqc</code>, <code>multiqc</code></li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="c1"># Load modules</span>
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>module<span class="w"> </span>load<span class="w"> </span>fastqc
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>module<span class="w"> </span>load<span class="w"> </span>multiqc
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a><span class="nv">indata</span><span class="o">=</span><span class="s2">&quot;/data/users/user29/metagenomes/shotgun/&quot;</span>
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>
</span><span id="__span-1-6"><a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a><span class="c1">## Create Dir</span>
</span><span id="__span-1-7"><a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">outfastqc</span><span class="si">}</span><span class="w"> </span>/data/users/user24/metagenomes/shotgun/scripts<span class="w"> </span>/data/users/user24/metagenomes/shotgun/logs
</span><span id="__span-1-8"><a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a>
</span><span id="__span-1-9"><a id="__codelineno-1-9" name="__codelineno-1-9" href="#__codelineno-1-9"></a><span class="c1"># create a samplesheet.csv with three columns samplename,fastq_1,fastq_2</span>
</span><span id="__span-1-10"><a id="__codelineno-1-10" name="__codelineno-1-10" href="#__codelineno-1-10"></a>python<span class="w"> </span>/data/users/user24/metagenomes/shotgun/scripts/samplesheet_generator.py<span class="w"> </span><span class="si">${</span><span class="nv">indata</span><span class="si">}</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-1-11"><a id="__codelineno-1-11" name="__codelineno-1-11" href="#__codelineno-1-11"></a><span class="w">  </span>/data/users/<span class="si">${</span><span class="nv">USER</span><span class="si">}</span>/metagenomes/shotgun/samplesheet.csv
</span><span id="__span-1-12"><a id="__codelineno-1-12" name="__codelineno-1-12" href="#__codelineno-1-12"></a>
</span><span id="__span-1-13"><a id="__codelineno-1-13" name="__codelineno-1-13" href="#__codelineno-1-13"></a><span class="c1"># Run raw QC</span>
</span><span id="__span-1-14"><a id="__codelineno-1-14" name="__codelineno-1-14" href="#__codelineno-1-14"></a>nextflow<span class="w"> </span>run<span class="w"> </span>/data/users/user24/metagenomes/shotgun/scripts/qc_pipeline_v1.nf<span class="w"> </span><span class="se">\</span>
</span><span id="__span-1-15"><a id="__codelineno-1-15" name="__codelineno-1-15" href="#__codelineno-1-15"></a><span class="w">  </span>--input<span class="w"> </span>/data/users/<span class="si">${</span><span class="nv">USER</span><span class="si">}</span>/metagenomes/shotgun/samplesheet.csv<span class="w"> </span><span class="se">\</span>
</span><span id="__span-1-16"><a id="__codelineno-1-16" name="__codelineno-1-16" href="#__codelineno-1-16"></a><span class="w">  </span>--outdir<span class="w"> </span>/data/users/<span class="si">${</span><span class="nv">USER</span><span class="si">}</span>/metagenomes/shotgun/results/rawfastqc
</span></code></pre></div>
<p><strong>What to think about?</strong>
- Which parts of the data are flagged as potentially problematic? GC% content of the dataset.
NB: We are dealing with mixed community and organisms, so its difficult to  have a perfect normal distribution around the average.
As such we consider this as normal given the biology of the system.
- Does the sequencing length matches the libraries used? If sequences are shorter than expected, are adapters
 a concern?
- Are adapters and/or barcodes removed?
  - Look at the Per base sequence content to diagnose this.
- Is there unexpected sequence duplication?
  - This can occur when low-input library preparations are used.
- Are over-represented k-mers present?
  - This can be a sign of adapter and barcode contamination.</p>
<hr />
<h3 id="step-3-trimming-filtering">Step 3. Trimming &amp; Filtering<a class="headerlink" href="#step-3-trimming-filtering" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: Removes adapters (which can cause false alignments \&amp; false taxonomic assignments), low-quality bases (increase error rates), and very short 
reads (to standardize read lengths) that bias downstream analysis.</li>
<li><strong>Tool</strong>: <code>fastp</code>, <code>trimmomatic</code>, <code>BBMap</code>, <code>sickle</code>, <code>cutadapt</code>, etc.</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a><span class="c1"># Load required modules/tools</span>
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>module<span class="w"> </span>load<span class="w"> </span>trimmomatic
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a><span class="c1"># Define input and output dir</span>
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a><span class="nv">indata</span><span class="o">=</span><span class="s2">&quot;/data/users/user24/metagenomes/&quot;</span>
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a><span class="nv">wkdir</span><span class="o">=</span><span class="s2">&quot;/data/users/</span><span class="si">${</span><span class="nv">USER</span><span class="si">}</span><span class="s2">/metagenomics/shotgun/&quot;</span>
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a><span class="nv">outtrimmomatic</span><span class="o">=</span><span class="si">${</span><span class="nv">wkdir</span><span class="si">}</span><span class="s2">&quot;/data_analysis/02_trimmomatic&quot;</span>
</span><span id="__span-2-10"><a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a>
</span><span id="__span-2-11"><a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}</span>
</span><span id="__span-2-12"><a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a>
</span><span id="__span-2-13"><a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a><span class="c1"># Trimming with Trimmomatic</span>
</span><span id="__span-2-14"><a id="__codelineno-2-14" name="__codelineno-2-14" href="#__codelineno-2-14"></a><span class="k">for</span><span class="w"> </span>file<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="sb">`</span>ls<span class="w"> </span><span class="si">${</span><span class="nv">data</span><span class="si">}</span>*1.fastq.gz<span class="sb">`</span>
</span><span id="__span-2-15"><a id="__codelineno-2-15" name="__codelineno-2-15" href="#__codelineno-2-15"></a><span class="k">do</span>
</span><span id="__span-2-16"><a id="__codelineno-2-16" name="__codelineno-2-16" href="#__codelineno-2-16"></a><span class="w">  </span><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">file</span><span class="si">}</span><span class="w"> </span>_1.fastq.gz<span class="k">)</span>
</span><span id="__span-2-17"><a id="__codelineno-2-17" name="__codelineno-2-17" href="#__codelineno-2-17"></a><span class="w">  </span>trimmomatic<span class="w"> </span>PE<span class="w"> </span>-threads<span class="w"> </span><span class="m">8</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-2-18"><a id="__codelineno-2-18" name="__codelineno-2-18" href="#__codelineno-2-18"></a><span class="w">      </span><span class="si">${</span><span class="nv">indata</span><span class="si">}${</span><span class="nv">sample</span><span class="si">}</span>_R1.fastq.gz<span class="w"> </span><span class="o">{</span>indata<span class="o">}</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_R2.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-2-19"><a id="__codelineno-2-19" name="__codelineno-2-19" href="#__codelineno-2-19"></a><span class="w">      </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}${</span><span class="nv">sample</span><span class="si">}</span>_R1.fastq.gz<span class="w"> </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}${</span><span class="nv">sample</span><span class="si">}</span>_R1_unpaired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-2-20"><a id="__codelineno-2-20" name="__codelineno-2-20" href="#__codelineno-2-20"></a><span class="w">      </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}${</span><span class="nv">sample</span><span class="si">}</span>_R2.fastq.gz<span class="w"> </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}${</span><span class="nv">sample</span><span class="si">}</span>_R2_unpaired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-2-21"><a id="__codelineno-2-21" name="__codelineno-2-21" href="#__codelineno-2-21"></a><span class="w">      </span>ILLUMINACLIP:adapters.fa:2:30:10<span class="w"> </span><span class="se">\</span>
</span><span id="__span-2-22"><a id="__codelineno-2-22" name="__codelineno-2-22" href="#__codelineno-2-22"></a><span class="w">      </span>LEADING:3<span class="w"> </span>TRAILING:3<span class="w"> </span><span class="se">\</span>
</span><span id="__span-2-23"><a id="__codelineno-2-23" name="__codelineno-2-23" href="#__codelineno-2-23"></a><span class="w">      </span>SLIDINGWINDOW:4:20<span class="w"> </span><span class="se">\</span>
</span><span id="__span-2-24"><a id="__codelineno-2-24" name="__codelineno-2-24" href="#__codelineno-2-24"></a><span class="w">      </span>MINLEN:50
</span><span id="__span-2-25"><a id="__codelineno-2-25" name="__codelineno-2-25" href="#__codelineno-2-25"></a><span class="k">done</span>
</span><span id="__span-2-26"><a id="__codelineno-2-26" name="__codelineno-2-26" href="#__codelineno-2-26"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Trimming with Trimmomatic completed&quot;</span>
</span></code></pre></div>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>PE</td>
<td>positional</td>
<td>Specifies whether we are analysing single- or paired-end reads</td>
</tr>
<tr>
<td>-threads 2</td>
<td>keyword</td>
<td>Specifies the number of threads to use when processing</td>
</tr>
<tr>
<td>-phred33</td>
<td>keyword</td>
<td>Specifies the fastq encoding used</td>
</tr>
<tr>
<td><span class="arithmatex" sample="sample">\({indata}\)</span>_R2.fastq.gz}_R1.fastq.gz {indata}${sample</td>
<td>positional</td>
<td>The paired forward and reverse reads to trim</td>
</tr>
<tr>
<td>ILLUMINACLIP:adapters.fa:2:30:10</td>
<td>positional</td>
<td>Adapter trimming allowing for 2 seed mismatch, palindrome clip score threshold of 30, and simple clip score threshold of 10</td>
</tr>
<tr>
<td>SLIDINGWINDOW:4:20</td>
<td>positional</td>
<td>Quality filtering command. Analyse each sequence in a 4 base pair sliding window and then truncate if the average quality drops below Q20</td>
</tr>
<tr>
<td>MINLEN:50</td>
<td>positional</td>
<td>Length filtering command. Discard sequences that are shorter than 80 base pairs after trimming</td>
</tr>
</tbody>
</table>
<blockquote>
<p><strong>Note:</strong> The trimming parameters in <code>trimmomatic</code> are processed in the order they are specified. For instance, </p>
</blockquote>
<hr />
<p><div class="language-bash highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="k">for</span><span class="w"> </span>file<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="sb">`</span>ls<span class="w"> </span><span class="si">${</span><span class="nv">data</span><span class="si">}</span>*1.fastq.gz<span class="sb">`</span>
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a><span class="k">do</span>
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="w">  </span><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">file</span><span class="si">}</span><span class="w"> </span>_1.fastq.gz<span class="k">)</span>
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="w">  </span>trimmomatic<span class="w"> </span>PE<span class="w"> </span>-threads<span class="w"> </span><span class="m">8</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a><span class="w">        </span><span class="si">${</span><span class="nv">indata</span><span class="si">}${</span><span class="nv">sample</span><span class="si">}</span>_R1.fastq.gz<span class="w"> </span><span class="o">{</span>indata<span class="o">}</span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span>_R2.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-6"><a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a><span class="w">        </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}${</span><span class="nv">sample</span><span class="si">}</span>_R1.fastq.gz<span class="w"> </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}${</span><span class="nv">sample</span><span class="si">}</span>_R1_unpaired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-7"><a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a><span class="w">        </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}${</span><span class="nv">sample</span><span class="si">}</span>_R2.fastq.gz<span class="w"> </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}${</span><span class="nv">sample</span><span class="si">}</span>_R2_unpaired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-8"><a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a><span class="w">        </span>ILLUMINACLIP:adapters.fa:2:30:10<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-9"><a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a><span class="w">        </span>LEADING:3<span class="w"> </span>TRAILING:3<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-10"><a id="__codelineno-3-10" name="__codelineno-3-10" href="#__codelineno-3-10"></a><span class="w">        </span>MINLEN:50<span class="w"> </span><span class="se">\</span>
</span><span id="__span-3-11"><a id="__codelineno-3-11" name="__codelineno-3-11" href="#__codelineno-3-11"></a><span class="w">        </span>SLIDINGWINDOW:4:20<span class="w"> </span>
</span><span id="__span-3-12"><a id="__codelineno-3-12" name="__codelineno-3-12" href="#__codelineno-3-12"></a><span class="k">done</span>
</span></code></pre></div>
means we remove sequences shorter than 50 bps and then qualiyty trim, thus if a sequence is trimmed to a length shorter than 50bps after trimming, the <code>MINLEN</code> filtering does not execute a second time.</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Delete unnecessary files</span>
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>rm<span class="w"> </span>-rf<span class="w"> </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}</span>*<span class="si">${</span><span class="nv">sample</span><span class="si">}</span>*_unpaired.fastq.gz
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="c1"># Quality checking</span>
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>module<span class="w"> </span>load<span class="w"> </span>fastqc<span class="w"> </span>multiqc
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>fastqc<span class="w"> </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}</span>*<span class="w"> </span>-o<span class="w"> </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}</span>
</span><span id="__span-4-7"><a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a>multiqc<span class="w">  </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}</span><span class="w"> </span>-o<span class="w">  </span><span class="si">${</span><span class="nv">outtrimmomatic</span><span class="si">}</span>
</span></code></pre></div>
<hr />
<h3 id="step-4-deduplication-host-dna-removal">Step 4. Deduplication &amp; Host DNA Removal<a class="headerlink" href="#step-4-deduplication-host-dna-removal" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>:</li>
<li>Often metagenomes are obtained from host-associated microbial communities. As a result, they contain significant amount of host DNA which may interfere with microbial analysis
  and create privacy concerns.</li>
<li>Specifically any studies invloving human subjects or samples derived from Taonga species.</li>
<li>
<p>Although several approaches are used for this, the most popular is to map reads to a reference genome (includin human genome). That is remove all reads that map to the reference of the dataset.</p>
</li>
<li>
<p><strong>Tools</strong>: <code>clumpify</code> (dedup), <code>bowtie2</code> or <code>bwa mem</code> (host removal).</p>
</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a><span class="nb">set</span><span class="w"> </span>-e<span class="w">  </span><span class="c1"># Exit on error</span>
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>
</span><span id="__span-5-4"><a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a><span class="c1"># Define dirs</span>
</span><span id="__span-5-5"><a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a><span class="nv">genome_dir</span><span class="o">=</span><span class="s2">&quot;/data/users/user24/refs/human_reference/&quot;</span>
</span><span id="__span-5-6"><a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a>
</span><span id="__span-5-7"><a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a><span class="c1">## Remove human contamination using BWA and SAMtools</span>
</span><span id="__span-5-8"><a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a><span class="c1">## Create dir</span>
</span><span id="__span-5-9"><a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">genome_dir</span><span class="si">}</span>
</span><span id="__span-5-10"><a id="__codelineno-5-10" name="__codelineno-5-10" href="#__codelineno-5-10"></a><span class="c1"># Download the Human refence genome</span>
</span><span id="__span-5-11"><a id="__codelineno-5-11" name="__codelineno-5-11" href="#__codelineno-5-11"></a><span class="nb">cd</span><span class="w"> </span><span class="si">${</span><span class="nv">genome_dir</span><span class="si">}</span>
</span><span id="__span-5-12"><a id="__codelineno-5-12" name="__codelineno-5-12" href="#__codelineno-5-12"></a>
</span><span id="__span-5-13"><a id="__codelineno-5-13" name="__codelineno-5-13" href="#__codelineno-5-13"></a><span class="c1"># Configuration</span>
</span><span id="__span-5-14"><a id="__codelineno-5-14" name="__codelineno-5-14" href="#__codelineno-5-14"></a><span class="nv">GENOME_VERSION</span><span class="o">=</span><span class="s2">&quot;GRCh38&quot;</span>
</span><span id="__span-5-15"><a id="__codelineno-5-15" name="__codelineno-5-15" href="#__codelineno-5-15"></a><span class="nv">BASE_URL</span><span class="o">=</span><span class="s2">&quot;https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38&quot;</span>
</span><span id="__span-5-16"><a id="__codelineno-5-16" name="__codelineno-5-16" href="#__codelineno-5-16"></a>
</span><span id="__span-5-17"><a id="__codelineno-5-17" name="__codelineno-5-17" href="#__codelineno-5-17"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;=== Downloading Human Reference Genome (</span><span class="si">${</span><span class="nv">GENOME_VERSION</span><span class="si">}</span><span class="s2">) ===&quot;</span>
</span><span id="__span-5-18"><a id="__codelineno-5-18" name="__codelineno-5-18" href="#__codelineno-5-18"></a>
</span><span id="__span-5-19"><a id="__codelineno-5-19" name="__codelineno-5-19" href="#__codelineno-5-19"></a><span class="c1"># Option 1: Download complete genome assembly (recommended for most applications)</span>
</span><span id="__span-5-20"><a id="__codelineno-5-20" name="__codelineno-5-20" href="#__codelineno-5-20"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Downloading complete genome assembly...&quot;</span>
</span><span id="__span-5-21"><a id="__codelineno-5-21" name="__codelineno-5-21" href="#__codelineno-5-21"></a>wget<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;</span><span class="si">${</span><span class="nv">BASE_URL</span><span class="si">}</span><span class="s2">/GCA_000001405.15_GRCh38_genomic.fna.gz&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-22"><a id="__codelineno-5-22" name="__codelineno-5-22" href="#__codelineno-5-22"></a><span class="w">     </span>-O<span class="w"> </span><span class="s2">&quot;GRCh38_genomic.fna.gz&quot;</span>
</span><span id="__span-5-23"><a id="__codelineno-5-23" name="__codelineno-5-23" href="#__codelineno-5-23"></a><span class="c1"># Decompress</span>
</span><span id="__span-5-24"><a id="__codelineno-5-24" name="__codelineno-5-24" href="#__codelineno-5-24"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Decompressing genome file...&quot;</span>
</span><span id="__span-5-25"><a id="__codelineno-5-25" name="__codelineno-5-25" href="#__codelineno-5-25"></a>gunzip<span class="w"> </span>-f<span class="w"> </span>GRCh38_genomic.fna.gz
</span><span id="__span-5-26"><a id="__codelineno-5-26" name="__codelineno-5-26" href="#__codelineno-5-26"></a>
</span><span id="__span-5-27"><a id="__codelineno-5-27" name="__codelineno-5-27" href="#__codelineno-5-27"></a><span class="c1"># Option 2: Download chromosome-only version (excludes contigs/scaffolds)</span>
</span><span id="__span-5-28"><a id="__codelineno-5-28" name="__codelineno-5-28" href="#__codelineno-5-28"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Downloading chromosome-only version...&quot;</span>
</span><span id="__span-5-29"><a id="__codelineno-5-29" name="__codelineno-5-29" href="#__codelineno-5-29"></a>wget<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz&quot;</span><span class="w"> </span><span class="se">\</span>
</span><span id="__span-5-30"><a id="__codelineno-5-30" name="__codelineno-5-30" href="#__codelineno-5-30"></a><span class="w">     </span>-O<span class="w"> </span><span class="s2">&quot;hg38_chromosomes_only.fa.gz&quot;</span>
</span><span id="__span-5-31"><a id="__codelineno-5-31" name="__codelineno-5-31" href="#__codelineno-5-31"></a>
</span><span id="__span-5-32"><a id="__codelineno-5-32" name="__codelineno-5-32" href="#__codelineno-5-32"></a>gunzip<span class="w"> </span>-f<span class="w"> </span>hg38_chromosomes_only.fa.gz
</span></code></pre></div>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="ch">#!/bin/bash</span>
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a><span class="c1"># Load modules</span>
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>module<span class="w"> </span>load<span class="w"> </span>bwa
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>module<span class="w"> </span>load<span class="w"> </span>bowtie2
</span><span id="__span-6-6"><a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a>
</span><span id="__span-6-7"><a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;=== Building BWA Index ===&quot;</span>
</span><span id="__span-6-8"><a id="__codelineno-6-8" name="__codelineno-6-8" href="#__codelineno-6-8"></a><span class="c1"># Build BWA index for alignment (required for host removal, one time set-up)</span>
</span><span id="__span-6-9"><a id="__codelineno-6-9" name="__codelineno-6-9" href="#__codelineno-6-9"></a>bwa<span class="w"> </span>index<span class="w"> </span>GRCh38_genomic.fna
</span><span id="__span-6-10"><a id="__codelineno-6-10" name="__codelineno-6-10" href="#__codelineno-6-10"></a>
</span><span id="__span-6-11"><a id="__codelineno-6-11" name="__codelineno-6-11" href="#__codelineno-6-11"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;=== Verifying Download ===&quot;</span>
</span><span id="__span-6-12"><a id="__codelineno-6-12" name="__codelineno-6-12" href="#__codelineno-6-12"></a><span class="c1"># Check file integrity</span>
</span><span id="__span-6-13"><a id="__codelineno-6-13" name="__codelineno-6-13" href="#__codelineno-6-13"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Genome file size:&quot;</span>
</span><span id="__span-6-14"><a id="__codelineno-6-14" name="__codelineno-6-14" href="#__codelineno-6-14"></a>ls<span class="w"> </span>-lh<span class="w"> </span>*.fna
</span><span id="__span-6-15"><a id="__codelineno-6-15" name="__codelineno-6-15" href="#__codelineno-6-15"></a>
</span><span id="__span-6-16"><a id="__codelineno-6-16" name="__codelineno-6-16" href="#__codelineno-6-16"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Number of sequences:&quot;</span>
</span><span id="__span-6-17"><a id="__codelineno-6-17" name="__codelineno-6-17" href="#__codelineno-6-17"></a>grep<span class="w"> </span>-c<span class="w"> </span><span class="s2">&quot;&gt;&quot;</span><span class="w"> </span>GRCh38_genomic.fna
</span><span id="__span-6-18"><a id="__codelineno-6-18" name="__codelineno-6-18" href="#__codelineno-6-18"></a>
</span><span id="__span-6-19"><a id="__codelineno-6-19" name="__codelineno-6-19" href="#__codelineno-6-19"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;First few sequence headers:&quot;</span>
</span><span id="__span-6-20"><a id="__codelineno-6-20" name="__codelineno-6-20" href="#__codelineno-6-20"></a>grep<span class="w"> </span><span class="s2">&quot;&gt;&quot;</span><span class="w"> </span>GRCh38_genomic.fna<span class="w"> </span><span class="p">|</span><span class="w"> </span>head<span class="w"> </span>-10
</span><span id="__span-6-21"><a id="__codelineno-6-21" name="__codelineno-6-21" href="#__codelineno-6-21"></a>
</span><span id="__span-6-22"><a id="__codelineno-6-22" name="__codelineno-6-22" href="#__codelineno-6-22"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;=== Download Complete ===&quot;</span>
</span><span id="__span-6-23"><a id="__codelineno-6-23" name="__codelineno-6-23" href="#__codelineno-6-23"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Reference genome files are ready in: </span><span class="k">$(</span><span class="nb">pwd</span><span class="k">)</span><span class="s2">&quot;</span>
</span><span id="__span-6-24"><a id="__codelineno-6-24" name="__codelineno-6-24" href="#__codelineno-6-24"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Main genome file: GRCh38_genomic.fna&quot;</span>
</span><span id="__span-6-25"><a id="__codelineno-6-25" name="__codelineno-6-25" href="#__codelineno-6-25"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Chromosome-only file: hg38_chromosomes_only.fa&quot;</span>
</span><span id="__span-6-26"><a id="__codelineno-6-26" name="__codelineno-6-26" href="#__codelineno-6-26"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;&quot;</span>
</span><span id="__span-6-27"><a id="__codelineno-6-27" name="__codelineno-6-27" href="#__codelineno-6-27"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Files generated:&quot;</span>
</span><span id="__span-6-28"><a id="__codelineno-6-28" name="__codelineno-6-28" href="#__codelineno-6-28"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;- GRCh38_genomic.fna (main reference)&quot;</span>
</span><span id="__span-6-29"><a id="__codelineno-6-29" name="__codelineno-6-29" href="#__codelineno-6-29"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;- GRCh38_genomic.fna.amb, .ann, .bwt, .pac, .sa (BWA index)&quot;</span>
</span><span id="__span-6-30"><a id="__codelineno-6-30" name="__codelineno-6-30" href="#__codelineno-6-30"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;- GRCh38_genomic.fna.fai (samtools index)&quot;</span>
</span><span id="__span-6-31"><a id="__codelineno-6-31" name="__codelineno-6-31" href="#__codelineno-6-31"></a>
</span><span id="__span-6-32"><a id="__codelineno-6-32" name="__codelineno-6-32" href="#__codelineno-6-32"></a><span class="c1"># Align reads to human genome</span>
</span><span id="__span-6-33"><a id="__codelineno-6-33" name="__codelineno-6-33" href="#__codelineno-6-33"></a><span class="k">for</span><span class="w"> </span>file<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="sb">`</span>ls<span class="w"> </span><span class="si">${</span><span class="nv">indata</span><span class="si">}</span>*1.fastq.gz<span class="sb">`</span>
</span><span id="__span-6-34"><a id="__codelineno-6-34" name="__codelineno-6-34" href="#__codelineno-6-34"></a><span class="k">do</span>
</span><span id="__span-6-35"><a id="__codelineno-6-35" name="__codelineno-6-35" href="#__codelineno-6-35"></a><span class="w">  </span><span class="nv">sample</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">file</span><span class="si">}</span><span class="w"> </span>_1.fastq.gz<span class="k">)</span>
</span><span id="__span-6-36"><a id="__codelineno-6-36" name="__codelineno-6-36" href="#__codelineno-6-36"></a><span class="w">  </span>bowtie2<span class="w"> </span>-x<span class="w"> </span><span class="si">${</span><span class="nv">refs</span><span class="si">}</span>human_index<span class="w"> </span>-1<span class="w"> </span><span class="o">{</span>sample<span class="o">}</span><span class="si">${</span><span class="nv">data</span><span class="si">}</span>_trimmed_R1.fastq.gz<span class="w"> </span>-2<span class="w"> </span>sample_trimmed_R2.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-37"><a id="__codelineno-6-37" name="__codelineno-6-37" href="#__codelineno-6-37"></a><span class="w">    </span>--un-conc<span class="w"> </span>sample_host_removed.fastq.gz<span class="w"> </span>-S<span class="w"> </span>/dev/null
</span><span id="__span-6-38"><a id="__codelineno-6-38" name="__codelineno-6-38" href="#__codelineno-6-38"></a>
</span><span id="__span-6-39"><a id="__codelineno-6-39" name="__codelineno-6-39" href="#__codelineno-6-39"></a>
</span><span id="__span-6-40"><a id="__codelineno-6-40" name="__codelineno-6-40" href="#__codelineno-6-40"></a><span class="w">  </span>bwa<span class="w"> </span>mem<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>human_reference.fasta<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-41"><a id="__codelineno-6-41" name="__codelineno-6-41" href="#__codelineno-6-41"></a><span class="w">      </span>trimmed_R1_paired.fastq.gz<span class="w"> </span>trimmed_R2_paired.fastq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-6-42"><a id="__codelineno-6-42" name="__codelineno-6-42" href="#__codelineno-6-42"></a><span class="w">      </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>view<span class="w"> </span>-b<span class="w"> </span>-f<span class="w"> </span><span class="m">4</span><span class="w"> </span>-<span class="w"> </span>&gt;<span class="w"> </span>unmapped_reads.bam
</span><span id="__span-6-43"><a id="__codelineno-6-43" name="__codelineno-6-43" href="#__codelineno-6-43"></a>
</span><span id="__span-6-44"><a id="__codelineno-6-44" name="__codelineno-6-44" href="#__codelineno-6-44"></a><span class="w">  </span><span class="c1"># Convert unmapped reads back to FASTQ</span>
</span><span id="__span-6-45"><a id="__codelineno-6-45" name="__codelineno-6-45" href="#__codelineno-6-45"></a><span class="w">  </span>samtools<span class="w"> </span>fastq<span class="w"> </span>-1<span class="w"> </span>dehosted_R1.fastq.gz<span class="w"> </span>-2<span class="w"> </span>dehosted_R2.fastq.gz<span class="w"> </span>unmapped_reads.bam
</span><span id="__span-6-46"><a id="__codelineno-6-46" name="__codelineno-6-46" href="#__codelineno-6-46"></a>
</span><span id="__span-6-47"><a id="__codelineno-6-47" name="__codelineno-6-47" href="#__codelineno-6-47"></a><span class="nb">echo</span><span class="w"> </span><span class="s2">&quot;Host removal completed&quot;</span>
</span></code></pre></div>
<hr />
<h3 id="step-5-taxonomic-profiling-read-based">Step 5. Taxonomic Profiling (Read-Based)<a class="headerlink" href="#step-5-taxonomic-profiling-read-based" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: Directly assigns taxonomy without assembly, faster and less computationally intensive.</li>
<li><strong>Tools</strong>: Kraken2 + Bracken, MetaPhlAn</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>kraken2<span class="w"> </span>--db<span class="w"> </span>kraken2_db<span class="w"> </span>--paired<span class="w"> </span>sample_host_removed.1.fq.gz<span class="w"> </span>sample_host_removed.2.fq.gz<span class="w"> </span><span class="se">\</span>
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="w">  </span>--output<span class="w"> </span>kraken2_output.txt<span class="w"> </span>--report<span class="w"> </span>kraken2_report.txt
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>bracken<span class="w"> </span>-d<span class="w"> </span>kraken2_db<span class="w"> </span>-i<span class="w"> </span>kraken2_report.txt<span class="w"> </span>-o<span class="w"> </span>bracken_species.txt<span class="w"> </span>-r<span class="w"> </span><span class="m">150</span><span class="w"> </span>-l<span class="w"> </span>S
</span></code></pre></div>
<hr />
<h3 id="step-6-functional-profiling-read-based">Step 6. Functional Profiling (Read-Based)<a class="headerlink" href="#step-6-functional-profiling-read-based" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: Identifies pathways/genes present without needing assembly.</li>
<li><strong>Tool</strong>: HUMAnN</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>humann<span class="w"> </span>--input<span class="w"> </span>sample_host_removed.fastq.gz<span class="w"> </span>--output<span class="w"> </span>humann_out/
</span></code></pre></div>
<hr />
<h3 id="step-7-assembly-based-profiling">Step 7. Assembly-Based Profiling<a class="headerlink" href="#step-7-assembly-based-profiling" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: Reconstructs contigs/MAGs b enables strain-level analysis, discovery of new genes, plasmids, MGEs.</li>
<li><strong>Tools</strong>: <strong>MEGAHIT</strong> or <strong>metaSPAdes</strong></li>
</ul>
<h4 id="megahit">MEGAHIT<a class="headerlink" href="#megahit" title="Permanent link">&para;</a></h4>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>megahit<span class="w"> </span>-1<span class="w"> </span>sample_host_removed.1.fq.gz<span class="w"> </span>-2<span class="w"> </span>sample_host_removed.2.fq.gz<span class="w"> </span>-o<span class="w"> </span>megahit_out/
</span></code></pre></div>
<h4 id="metaspades">metaSPAdes<a class="headerlink" href="#metaspades" title="Permanent link">&para;</a></h4>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>spades.py<span class="w"> </span>--meta<span class="w"> </span>-1<span class="w"> </span>sample_host_removed.1.fq.gz<span class="w"> </span>-2<span class="w"> </span>sample_host_removed.2.fq.gz<span class="w"> </span>-o<span class="w"> </span>metaspades_out/
</span></code></pre></div>
<p><strong>MEGAHIT vs metaSPAdes:</strong></p>
<ul>
<li><strong>MEGAHIT Advantages:</strong></li>
<li>Extremely fast, lower memory usage.</li>
<li>Scales better for very large datasets (e.g., population metagenomes).</li>
<li><strong>MEGAHIT Disadvantages:</strong></li>
<li>May produce slightly shorter contigs than metaSPAdes.</li>
<li><strong>metaSPAdes Advantages:</strong></li>
<li>Produces higher-quality, longer assemblies (useful for MAG recovery).</li>
<li><strong>metaSPAdes Disadvantages:</strong></li>
<li>Requires more RAM and CPU time.</li>
</ul>
<hr />
<h3 id="step-8-mapping-binning">Step 8. Mapping &amp; Binning<a class="headerlink" href="#step-8-mapping-binning" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: Group contigs into MAGs, quantify abundances.</li>
<li><strong>Tools</strong>: <code>bowtie2</code>, <code>samtools</code>, <code>MetaBAT2</code></li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>bowtie2<span class="w"> </span>-x<span class="w"> </span>megahit_out/final.contigs.fa<span class="w"> </span>-1<span class="w"> </span>sample_host_removed.1.fq.gz<span class="w"> </span>-2<span class="w"> </span>sample_host_removed.2.fq.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-o<span class="w"> </span>aln.bam
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>metabat2<span class="w"> </span>-i<span class="w"> </span>megahit_out/final.contigs.fa<span class="w"> </span>-a<span class="w"> </span>depth.txt<span class="w"> </span>-o<span class="w"> </span>bins_dir/bin
</span></code></pre></div>
<hr />
<h3 id="step-9-mag-quality-control">Step 9. MAG Quality Control<a class="headerlink" href="#step-9-mag-quality-control" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: Ensures completeness &amp; contamination are acceptable.</li>
<li><strong>Tool</strong>: CheckM</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>checkm<span class="w"> </span>lineage_wf<span class="w"> </span>bins_dir/<span class="w"> </span>checkm_out/
</span></code></pre></div>
<hr />
<h3 id="step-10-annotation-specialized-analyses">Step 10. Annotation &amp; Specialized Analyses<a class="headerlink" href="#step-10-annotation-specialized-analyses" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: Identify AMR, virulence, plasmids, metabolic capacity.</li>
<li><strong>Tools</strong>: Prokka, Bakta, AMRFinderPlus, ABRicate, DRAM</li>
</ul>
<h4 id="where-dram-fits-in">Where DRAM fits in:<a class="headerlink" href="#where-dram-fits-in" title="Permanent link">&para;</a></h4>
<ul>
<li><strong>Swap in DRAM</strong> in place of Prokka/Bakta for <strong>functional annotation of MAGs</strong>.</li>
<li>DRAM produces detailed metabolic profiles, pathway reconstruction, and microbial ecology insights.</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>DRAM.py<span class="w"> </span>annotate<span class="w"> </span>-i<span class="w"> </span>bins_dir/<span class="w"> </span>-o<span class="w"> </span>dram_out/<span class="w"> </span>--threads<span class="w"> </span><span class="m">16</span>
</span></code></pre></div>
<hr />
<h3 id="step-11-abundance-estimation-visualization">Step 11. Abundance Estimation &amp; Visualization<a class="headerlink" href="#step-11-abundance-estimation-visualization" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: Quantifies taxa/genes b links to clinical or epidemiological metadata.</li>
<li><strong>Tools</strong>: CoverM, Krona, R for plots.</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a>coverm<span class="w"> </span>contig<span class="w"> </span>--bam-files<span class="w"> </span>aln.bam<span class="w"> </span>--reference<span class="w"> </span>megahit_out/final.contigs.fa<span class="w"> </span>--methods<span class="w"> </span>tpm<span class="w"> </span>&gt;<span class="w"> </span>coverm_tpm.tsv
</span></code></pre></div>
<hr />
<h3 id="step-12-reporting-reproducibility">Step 12. Reporting &amp; Reproducibility<a class="headerlink" href="#step-12-reporting-reproducibility" title="Permanent link">&para;</a></h3>
<ul>
<li><strong>Why</strong>: Essential for public health applications.</li>
<li>Use <strong>Nextflow + Singularity/Conda</strong> for reproducible pipelines.</li>
<li>Summarize results in Excel or RMarkdown reports.</li>
</ul>
<hr />
<h3 id="nextflow-pipelines">Nextflow pipelines<a class="headerlink" href="#nextflow-pipelines" title="Permanent link">&para;</a></h3>
<h4 id="nfcoremag-pipeline-httpsnf-coremag400">nfcore/mag Pipeline <a href="https://nf-co.re/mag/4.0.0">https://nf-co.re/mag/4.0.0</a><a class="headerlink" href="#nfcoremag-pipeline-httpsnf-coremag400" title="Permanent link">&para;</a></h4>
<ul>
<li>Use an <a href="https://nf-co.re/mag/4.0.0/">nf-core/mag</a> pipeline for assembly, binning and annotation of metagenomes, <a href="https://github.com/nf-core/mag/tree/4.0.0">github repository</a>.</li>
<li>This pipeline works for short- and/or long-reads.</li>
<li><strong>Key Features:</strong></li>
<li><strong>Preprocessing:</strong><ul>
<li>Short reads: fastp, Bowtie2, FastQC</li>
<li>Long reads: Porechop, NanoLyse, Filtlong, NanoPlot</li>
</ul>
</li>
<li><strong>Assembly:</strong><ul>
<li>Short reads: MEGAHIT, SPAdes</li>
<li>Hybrid: hybridSPAdes</li>
</ul>
</li>
<li><strong>Binning:</strong><ul>
<li>Tools: MetaBAT2, MaxBin2, CONCOCT, DAS Tool</li>
<li>Quality checks: BUSCO, CheckM, GUNC</li>
</ul>
</li>
<li><strong>Taxonomic Classification:</strong><ul>
<li>Tools: GTDB-Tk, CAT/BAT</li>
<li>Co-assembly and co-abundance:</li>
<li>Supports sample grouping for co-assembly and binning</li>
</ul>
</li>
<li> <strong>Reproducibility:</strong></li>
<li>Uses Nextflow DSL2, Docker/Singularity containers</li>
<li>Fully portable across HPC, cloud, and local systems</li>
<li>Includes test datasets and CI testing</li>
<li>It requires a sample sheet in <code>csv</code>  format with five columns: sample,group,short_reads_1,short_reads_2,long_reads</li>
<li>Assuming all your raw reads (short- and long-reads) are in the same folder, run the python script:</li>
</ul>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="nv">proj</span><span class="o">=</span><span class="s2">&quot;/data/users/</span><span class="si">${</span><span class="nv">USER</span><span class="si">}</span><span class="s2">/metagenomes/shotgun/&quot;</span>
</span><span id="__span-15-2"><a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a><span class="c1"># Create required directories</span>
</span><span id="__span-15-3"><a id="__codelineno-15-3" name="__codelineno-15-3" href="#__codelineno-15-3"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">proj</span><span class="si">}</span>scripts<span class="w"> </span><span class="si">${</span><span class="nv">proj</span><span class="si">}</span>logs
</span><span id="__span-15-4"><a id="__codelineno-15-4" name="__codelineno-15-4" href="#__codelineno-15-4"></a><span class="c1"># Get the script to create samplesheet</span>
</span><span id="__span-15-5"><a id="__codelineno-15-5" name="__codelineno-15-5" href="#__codelineno-15-5"></a>cp<span class="w"> </span>/data/users/user24/metagenomes/shotgun/scripts/generate_mag_samplesheet.py<span class="w"> </span>/data/users/<span class="si">${</span><span class="nv">USER</span><span class="si">}</span>/metagenomes/shotgun/scripts/
</span><span id="__span-15-6"><a id="__codelineno-15-6" name="__codelineno-15-6" href="#__codelineno-15-6"></a>
</span><span id="__span-15-7"><a id="__codelineno-15-7" name="__codelineno-15-7" href="#__codelineno-15-7"></a><span class="c1"># Create samplesheet for running mag</span>
</span><span id="__span-15-8"><a id="__codelineno-15-8" name="__codelineno-15-8" href="#__codelineno-15-8"></a>python3<span class="w"> </span>/data/users/<span class="si">${</span><span class="nv">USER</span><span class="si">}</span>/metagenomes/shotgun/scripts/generate_mag_samplesheet.py<span class="w"> </span>/data/users/user29/metagenomes/shotgun/<span class="w"> </span>mag-samplesheet.csv
</span><span id="__span-15-9"><a id="__codelineno-15-9" name="__codelineno-15-9" href="#__codelineno-15-9"></a><span class="c1"># Create submission script</span>
</span><span id="__span-15-10"><a id="__codelineno-15-10" name="__codelineno-15-10" href="#__codelineno-15-10"></a>nano<span class="w"> </span>data/users/<span class="si">${</span><span class="nv">USER</span><span class="si">}</span>/metagenomes/shotgun/scripts/mag-nf_submit.sh
</span><span id="__span-15-11"><a id="__codelineno-15-11" name="__codelineno-15-11" href="#__codelineno-15-11"></a>
</span><span id="__span-15-12"><a id="__codelineno-15-12" name="__codelineno-15-12" href="#__codelineno-15-12"></a><span class="c1">#!/bin/bash</span>
</span><span id="__span-15-13"><a id="__codelineno-15-13" name="__codelineno-15-13" href="#__codelineno-15-13"></a><span class="c1">#SBATCH --job-name=&#39;mag&#39;</span>
</span><span id="__span-15-14"><a id="__codelineno-15-14" name="__codelineno-15-14" href="#__codelineno-15-14"></a><span class="c1">#SBATCH --time=24:00:00</span>
</span><span id="__span-15-15"><a id="__codelineno-15-15" name="__codelineno-15-15" href="#__codelineno-15-15"></a><span class="c1">#SBATCH --mem=128g</span>
</span><span id="__span-15-16"><a id="__codelineno-15-16" name="__codelineno-15-16" href="#__codelineno-15-16"></a><span class="c1">#SBATCH --ntasks=16</span>
</span><span id="__span-15-17"><a id="__codelineno-15-17" name="__codelineno-15-17" href="#__codelineno-15-17"></a><span class="c1">#SBATCH --output=/data/users/user24/metagenomes/shotgun/logs/nfcore-mag-stdout.log</span>
</span><span id="__span-15-18"><a id="__codelineno-15-18" name="__codelineno-15-18" href="#__codelineno-15-18"></a><span class="c1">#SBATCH --error=/data/users/user24/metagenomes/shotgun/logs/nfcore-mag-stderr.log</span>
</span><span id="__span-15-19"><a id="__codelineno-15-19" name="__codelineno-15-19" href="#__codelineno-15-19"></a><span class="c1">#SBATCH --mail-user=ephie.geza@uct.ac.za</span>
</span><span id="__span-15-20"><a id="__codelineno-15-20" name="__codelineno-15-20" href="#__codelineno-15-20"></a>
</span><span id="__span-15-21"><a id="__codelineno-15-21" name="__codelineno-15-21" href="#__codelineno-15-21"></a><span class="nv">proj</span><span class="o">=</span><span class="s2">&quot;/data/users/user24/metagenomes/shotgun/&quot;</span>
</span><span id="__span-15-22"><a id="__codelineno-15-22" name="__codelineno-15-22" href="#__codelineno-15-22"></a>
</span><span id="__span-15-23"><a id="__codelineno-15-23" name="__codelineno-15-23" href="#__codelineno-15-23"></a>module<span class="w"> </span>load<span class="w"> </span>nextflow/25.04.6
</span><span id="__span-15-24"><a id="__codelineno-15-24" name="__codelineno-15-24" href="#__codelineno-15-24"></a><span class="c1">#### Unload JAVA 18 as it doesn&#39;t work and load JAVA 17</span>
</span><span id="__span-15-25"><a id="__codelineno-15-25" name="__codelineno-15-25" href="#__codelineno-15-25"></a>module<span class="w"> </span>unload<span class="w"> </span>java/openjdk-18.0.2
</span><span id="__span-15-26"><a id="__codelineno-15-26" name="__codelineno-15-26" href="#__codelineno-15-26"></a>module<span class="w"> </span>load<span class="w"> </span>java/openjdk-17.0.2
</span><span id="__span-15-27"><a id="__codelineno-15-27" name="__codelineno-15-27" href="#__codelineno-15-27"></a><span class="c1">####Unset conflicting environment variables (optional but recommended)</span>
</span><span id="__span-15-28"><a id="__codelineno-15-28" name="__codelineno-15-28" href="#__codelineno-15-28"></a><span class="nb">unset</span><span class="w"> </span>JAVA_CMD
</span><span id="__span-15-29"><a id="__codelineno-15-29" name="__codelineno-15-29" href="#__codelineno-15-29"></a><span class="nb">unset</span><span class="w"> </span>JAVA_HOME
</span><span id="__span-15-30"><a id="__codelineno-15-30" name="__codelineno-15-30" href="#__codelineno-15-30"></a>
</span><span id="__span-15-31"><a id="__codelineno-15-31" name="__codelineno-15-31" href="#__codelineno-15-31"></a><span class="c1">#### Run pipeline</span>
</span><span id="__span-15-32"><a id="__codelineno-15-32" name="__codelineno-15-32" href="#__codelineno-15-32"></a>nextflow<span class="w"> </span>run<span class="w"> </span><span class="si">${</span><span class="nv">proj</span><span class="si">}</span>mag<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-33"><a id="__codelineno-15-33" name="__codelineno-15-33" href="#__codelineno-15-33"></a><span class="w">      </span>--input<span class="w"> </span><span class="si">${</span><span class="nv">proj</span><span class="si">}</span>mag-samplesheet.csv<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-34"><a id="__codelineno-15-34" name="__codelineno-15-34" href="#__codelineno-15-34"></a><span class="w">      </span>--outdir<span class="w"> </span><span class="si">${</span><span class="nv">proj</span><span class="si">}</span>nfcore-mag<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-35"><a id="__codelineno-15-35" name="__codelineno-15-35" href="#__codelineno-15-35"></a><span class="w">      </span>-w<span class="w"> </span><span class="si">${</span><span class="nv">work</span><span class="si">}</span>work/nfcore-mag<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-36"><a id="__codelineno-15-36" name="__codelineno-15-36" href="#__codelineno-15-36"></a><span class="w">      </span>-profile<span class="w"> </span>singularity<span class="w"> </span><span class="se">\</span>
</span><span id="__span-15-37"><a id="__codelineno-15-37" name="__codelineno-15-37" href="#__codelineno-15-37"></a><span class="w">      </span>-resume<span class="w"> </span>--skip_gtdbtk
</span><span id="__span-15-38"><a id="__codelineno-15-38" name="__codelineno-15-38" href="#__codelineno-15-38"></a>
</span><span id="__span-15-39"><a id="__codelineno-15-39" name="__codelineno-15-39" href="#__codelineno-15-39"></a><span class="c1">## Save and submit</span>
</span></code></pre></div>
<h3 id="nf-corefuncscan"><a href="https://nf-co.re/funcscan/2.1.0/">nf-core/funcscan</a><a class="headerlink" href="#nf-corefuncscan" title="Permanent link">&para;</a></h3>
<p>Using contigs to screen for functional and natural gene sequences</p>
<h3 id="viralgenie-nf-coreviralmetagenome">Viralgenie <a href="https://nf-co.re/viralmetagenome/0.1.2/">nf-core/viralmetagenome</a><a class="headerlink" href="#viralgenie-nf-coreviralmetagenome" title="Permanent link">&para;</a></h3>
<h3 id="nf-coretaxprofiler"><a href="https://nf-co.re/taxprofiler/1.2.4/">nf-core/taxprofiler</a><a class="headerlink" href="#nf-coretaxprofiler" title="Permanent link">&para;</a></h3>
<h2 id="targeted-metagenomics">Targeted Metagenomics<a class="headerlink" href="#targeted-metagenomics" title="Permanent link">&para;</a></h2>
<h3 id="nf-coreampliseq"><a href="https://nf-co.re/ampliseq/2.14.0/">nf-core/ampliseq</a><a class="headerlink" href="#nf-coreampliseq" title="Permanent link">&para;</a></h3>









  




                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 CIDRI-Africa
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/CIDRI-Africa" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/CIDRIAfrica" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill="currentColor" d="M459.4 151.7c.3 4.5.3 9.1.3 13.6 0 138.7-105.6 298.6-298.6 298.6-59.5 0-114.7-17.2-161.1-47.1 8.4 1 16.6 1.3 25.3 1.3 49.1 0 94.2-16.6 130.3-44.8-46.1-1-84.8-31.2-98.1-72.8 6.5 1 13 1.6 19.8 1.6 9.4 0 18.8-1.3 27.6-3.6-48.1-9.7-84.1-52-84.1-103v-1.3c14 7.8 30.2 12.7 47.4 13.3-28.3-18.8-46.8-51-46.8-87.4 0-19.5 5.2-37.4 14.3-53C87.4 130.8 165 172.4 252.1 176.9c-1.6-7.8-2.6-15.9-2.6-24C249.5 95.1 296.3 48 354.4 48c30.2 0 57.5 12.7 76.7 33.1 23.7-4.5 46.5-13.3 66.6-25.3-7.8 24.4-24.4 44.8-46.1 57.8 21.1-2.3 41.6-8.1 60.4-16.2-14.3 20.8-32.2 39.3-52.6 54.3"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
      <div class="md-progress" data-md-component="progress" role="progressbar"></div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["announce.dismiss", "content.action.edit", "content.action.view", "content.code.annotate", "content.code.copy", "content.code.select", "content.tabs.link", "content.tooltips", "header.autohide", "navigation.expand", "navigation.footer", "navigation.indexes", "navigation.instant", "navigation.instant.prefetch", "navigation.instant.progress", "navigation.prune", "navigation.sections", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow", "toc.integrate"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.92b07e13.min.js"></script>
      
        <script src="../../javascripts/extra.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
        <script src="https://unpkg.com/mermaid@10.6.1/dist/mermaid.min.js"></script>
      
        <script src="../../javascripts/mermaid.js"></script>
      
    
  </body>
</html>