<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="Hands-on training course for microbial genomics and metagenomics in clinical and public health applications"><meta name=author content=CIDRI-Africa><link href=https://cidri-africa.github.io/microbial-genomics-training/modules/day9/ rel=canonical><link href=../day8/ rel=prev><link href=../day10/ rel=next><link rel=icon href=../../assets/favicon.png><meta name=generator content="mkdocs-1.6.1, mkdocs-material-9.6.18"><title>Day 9 - Bring Your Own Data - Microbial Genomics & Metagenomics Training</title><link rel=stylesheet href=../../assets/stylesheets/main.7e37652d.min.css><link rel=stylesheet href=../../assets/stylesheets/palette.06af60db.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><link rel=stylesheet href=../../stylesheets/extra.css><script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script id=__analytics>function __md_analytics(){function e(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],e("js",new Date),e("config",""),document.addEventListener("DOMContentLoaded",(function(){document.forms.search&&document.forms.search.query.addEventListener("blur",(function(){this.value&&e("event","search",{search_term:this.value})}));document$.subscribe((function(){var t=document.forms.feedback;if(void 0!==t)for(var a of t.querySelectorAll("[type=submit]"))a.addEventListener("click",(function(a){a.preventDefault();var n=document.location.pathname,d=this.getAttribute("data-md-value");e("event","feedback",{page:n,data:d}),t.firstElementChild.disabled=!0;var r=t.querySelector(".md-feedback__note [data-md-value='"+d+"']");r&&(r.hidden=!1)})),t.hidden=!1})),location$.subscribe((function(t){e("config","",{page_path:t.pathname})}))}));var t=document.createElement("script");t.async=!0,t.src="https://www.googletagmanager.com/gtag/js?id=",document.getElementById("__analytics").insertAdjacentElement("afterEnd",t)}</script><script>"undefined"!=typeof __md_analytics&&__md_analytics()</script> <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary=teal data-md-color-accent=amber> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#day-9-bring-your-own-data class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class="md-header md-header--shadow md-header--lifted" data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../.. title="Microbial Genomics & Metagenomics Training" class="md-header__button md-logo" aria-label="Microbial Genomics & Metagenomics Training" data-md-component=logo> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 2h2v2c0 1.44.68 2.61 1.88 3.78.86.83 2.01 1.63 3.21 2.42l-1.83 1.19C8.27 10.72 7.31 10 6.5 9.21 5.07 7.82 4 6.1 4 4zm14 0h2v2c0 2.1-1.07 3.82-2.5 5.21-1.41 1.38-3.21 2.52-4.96 3.63-1.75 1.12-3.45 2.21-4.66 3.38C6.68 17.39 6 18.56 6 20v2H4v-2c0-2.1 1.07-3.82 2.5-5.21 1.41-1.38 3.21-2.52 4.96-3.63 1.75-1.12 3.45-2.21 4.66-3.38C17.32 6.61 18 5.44 18 4zm-3.26 10.61c.99.67 1.95 1.39 2.76 2.18C18.93 16.18 20 17.9 20 20v2h-2v-2c0-1.44-.68-2.61-1.88-3.78-.86-.83-2.01-1.63-3.21-2.42zM7 3h10v1l-.06.5H7.06L7 4zm.68 3h8.64c-.24.34-.52.69-.9 1.06l-.51.44H9.07l-.49-.44c-.38-.37-.66-.72-.9-1.06m1.41 10.5h5.84l.49.44c.38.37.66.72.9 1.06H7.68c.24-.34.52-.69.9-1.06zm-2.03 3h9.88l.06.5v1H7v-1z"/></svg> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Microbial Genomics & Metagenomics Training </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Day 9 - Bring Your Own Data </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=default data-md-color-primary=teal data-md-color-accent=amber aria-label="Switch to dark mode" type=radio name=__palette id=__palette_0> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=teal data-md-color-accent=amber aria-label="Switch to light mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_0 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg> </label> </form> <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" title=Share aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg> </a> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap tabindex=0 data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/CIDRI-Africa/microbial-genomics-training title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill=currentColor d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg> </div> <div class=md-source__repository> CIDRI-Africa/microbial-genomics-training </div> </a> </div> </nav> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../.. class=md-tabs__link> Home </a> </li> <li class=md-tabs__item> <a href=../../course/overview/ class=md-tabs__link> Course Information </a> </li> <li class="md-tabs__item md-tabs__item--active"> <a href=../day1/ class=md-tabs__link> Daily Modules </a> </li> <li class=md-tabs__item> <a href=../../datasets/ class=md-tabs__link> Resources </a> </li> </ul> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=../.. title="Microbial Genomics & Metagenomics Training" class="md-nav__button md-logo" aria-label="Microbial Genomics & Metagenomics Training" data-md-component=logo> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 2h2v2c0 1.44.68 2.61 1.88 3.78.86.83 2.01 1.63 3.21 2.42l-1.83 1.19C8.27 10.72 7.31 10 6.5 9.21 5.07 7.82 4 6.1 4 4zm14 0h2v2c0 2.1-1.07 3.82-2.5 5.21-1.41 1.38-3.21 2.52-4.96 3.63-1.75 1.12-3.45 2.21-4.66 3.38C6.68 17.39 6 18.56 6 20v2H4v-2c0-2.1 1.07-3.82 2.5-5.21 1.41-1.38 3.21-2.52 4.96-3.63 1.75-1.12 3.45-2.21 4.66-3.38C17.32 6.61 18 5.44 18 4zm-3.26 10.61c.99.67 1.95 1.39 2.76 2.18C18.93 16.18 20 17.9 20 20v2h-2v-2c0-1.44-.68-2.61-1.88-3.78-.86-.83-2.01-1.63-3.21-2.42zM7 3h10v1l-.06.5H7.06L7 4zm.68 3h8.64c-.24.34-.52.69-.9 1.06l-.51.44H9.07l-.49-.44c-.38-.37-.66-.72-.9-1.06m1.41 10.5h5.84l.49.44c.38.37.66.72.9 1.06H7.68c.24-.34.52-.69.9-1.06zm-2.03 3h9.88l.06.5v1H7v-1z"/></svg> </a> Microbial Genomics & Metagenomics Training </label> <div class=md-nav__source> <a href=https://github.com/CIDRI-Africa/microbial-genomics-training title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill=currentColor d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg> </div> <div class=md-source__repository> CIDRI-Africa/microbial-genomics-training </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. class=md-nav__link> <span class=md-ellipsis> Home </span> </a> </li> <li class="md-nav__item md-nav__item--pruned md-nav__item--nested"> <a href=../../course/overview/ class=md-nav__link> <span class=md-ellipsis> Course Information </span> <span class="md-nav__icon md-icon"></span> </a> </li> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_3 checked> <div class="md-nav__link md-nav__container"> <a href=../day1/ class="md-nav__link "> <span class=md-ellipsis> Daily Modules </span> </a> <label class="md-nav__link " for=__nav_3 id=__nav_3_label tabindex> <span class="md-nav__icon md-icon"></span> </label> </div> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_3_label aria-expanded=true> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Daily Modules </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../day2/ class=md-nav__link> <span class=md-ellipsis> Day 2 - Data Analysis </span> </a> </li> <li class=md-nav__item> <a href=../day3/ class=md-nav__link> <span class=md-ellipsis> Day 3 - Pathogen Genomics </span> </a> </li> <li class=md-nav__item> <a href=../day4/ class=md-nav__link> <span class=md-ellipsis> Day 4 - Comparative Genomics </span> </a> </li> <li class=md-nav__item> <a href=../day5/ class=md-nav__link> <span class=md-ellipsis> Day 5 - Metagenomic Profiling </span> </a> </li> <li class=md-nav__item> <a href=../day6/ class=md-nav__link> <span class=md-ellipsis> Day 6 - Co-infection & Mutations </span> </a> </li> <li class=md-nav__item> <a href=../day7/ class=md-nav__link> <span class=md-ellipsis> Day 7 - Nextflow </span> </a> </li> <li class=md-nav__item> <a href=../day8/ class=md-nav__link> <span class=md-ellipsis> Day 8 - Pipeline Development </span> </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> <span class=md-ellipsis> Day 9 - Bring Your Own Data </span> <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> <span class=md-ellipsis> Day 9 - Bring Your Own Data </span> </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#overview class=md-nav__link> <span class=md-ellipsis> Overview </span> </a> </li> <li class=md-nav__item> <a href=#learning-objectives class=md-nav__link> <span class=md-ellipsis> Learning Objectives </span> </a> </li> <li class=md-nav__item> <a href=#schedule class=md-nav__link> <span class=md-ellipsis> Schedule </span> </a> </li> <li class=md-nav__item> <a href=#session-structure class=md-nav__link> <span class=md-ellipsis> Session Structure </span> </a> <nav class=md-nav aria-label="Session Structure"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#opening-session-0900-0915 class=md-nav__link> <span class=md-ellipsis> Opening Session (09:00-09:15) </span> </a> </li> <li class=md-nav__item> <a href=#individual-analysis-time-0915-1130 class=md-nav__link> <span class=md-ellipsis> Individual Analysis Time (09:15-11:30) </span> </a> </li> <li class=md-nav__item> <a href=#progress-sharing-1145-1200 class=md-nav__link> <span class=md-ellipsis> Progress Sharing (11:45-12:00) </span> </a> </li> <li class=md-nav__item> <a href=#advanced-analysis-1200-1300 class=md-nav__link> <span class=md-ellipsis> Advanced Analysis (12:00-13:00) </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#data-types-and-analysis-approaches class=md-nav__link> <span class=md-ellipsis> Data Types and Analysis Approaches </span> </a> <nav class=md-nav aria-label="Data Types and Analysis Approaches"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#genomic-data class=md-nav__link> <span class=md-ellipsis> Genomic Data </span> </a> </li> <li class=md-nav__item> <a href=#metagenomic-data class=md-nav__link> <span class=md-ellipsis> Metagenomic Data </span> </a> </li> <li class=md-nav__item> <a href=#outbreak-investigation-data class=md-nav__link> <span class=md-ellipsis> Outbreak Investigation Data </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#technical-support-available class=md-nav__link> <span class=md-ellipsis> Technical Support Available </span> </a> <nav class=md-nav aria-label="Technical Support Available"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#computational-resources class=md-nav__link> <span class=md-ellipsis> Computational Resources </span> </a> </li> <li class=md-nav__item> <a href=#analysis-pipelines class=md-nav__link> <span class=md-ellipsis> Analysis Pipelines </span> </a> </li> <li class=md-nav__item> <a href=#expert-guidance class=md-nav__link> <span class=md-ellipsis> Expert Guidance </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#common-analysis-workflows class=md-nav__link> <span class=md-ellipsis> Common Analysis Workflows </span> </a> <nav class=md-nav aria-label="Common Analysis Workflows"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#genomic-surveillance-workflow class=md-nav__link> <span class=md-ellipsis> Genomic Surveillance Workflow </span> </a> </li> <li class=md-nav__item> <a href=#metagenomic-analysis-workflow class=md-nav__link> <span class=md-ellipsis> Metagenomic Analysis Workflow </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#troubleshooting-guide class=md-nav__link> <span class=md-ellipsis> Troubleshooting Guide </span> </a> <nav class=md-nav aria-label="Troubleshooting Guide"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#common-issues-and-solutions class=md-nav__link> <span class=md-ellipsis> Common Issues and Solutions </span> </a> <nav class=md-nav aria-label="Common Issues and Solutions"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#low-quality-data class=md-nav__link> <span class=md-ellipsis> Low-Quality Data </span> </a> </li> <li class=md-nav__item> <a href=#contamination-issues class=md-nav__link> <span class=md-ellipsis> Contamination Issues </span> </a> </li> <li class=md-nav__item> <a href=#assembly-problems class=md-nav__link> <span class=md-ellipsis> Assembly Problems </span> </a> </li> <li class=md-nav__item> <a href=#memoryresource-issues class=md-nav__link> <span class=md-ellipsis> Memory/Resource Issues </span> </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#analysis-documentation class=md-nav__link> <span class=md-ellipsis> Analysis Documentation </span> </a> <nav class=md-nav aria-label="Analysis Documentation"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#laboratory-notebook-template class=md-nav__link> <span class=md-ellipsis> Laboratory Notebook Template </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#resource-management class=md-nav__link> <span class=md-ellipsis> Resource Management </span> </a> <nav class=md-nav aria-label="Resource Management"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#data-organization class=md-nav__link> <span class=md-ellipsis> Data Organization </span> </a> </li> <li class=md-nav__item> <a href=#backup-strategies class=md-nav__link> <span class=md-ellipsis> Backup Strategies </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#collaboration-guidelines class=md-nav__link> <span class=md-ellipsis> Collaboration Guidelines </span> </a> <nav class=md-nav aria-label="Collaboration Guidelines"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#peer-support class=md-nav__link> <span class=md-ellipsis> Peer Support </span> </a> </li> <li class=md-nav__item> <a href=#trainer-consultation class=md-nav__link> <span class=md-ellipsis> Trainer Consultation </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#assessment-and-preparation-for-day-10 class=md-nav__link> <span class=md-ellipsis> Assessment and Preparation for Day 10 </span> </a> <nav class=md-nav aria-label="Assessment and Preparation for Day 10"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#presentation-preparation class=md-nav__link> <span class=md-ellipsis> Presentation Preparation </span> </a> </li> <li class=md-nav__item> <a href=#technical-documentation class=md-nav__link> <span class=md-ellipsis> Technical Documentation </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#success-metrics class=md-nav__link> <span class=md-ellipsis> Success Metrics </span> </a> </li> <li class=md-nav__item> <a href=#resources-for-continued-learning class=md-nav__link> <span class=md-ellipsis> Resources for Continued Learning </span> </a> <nav class=md-nav aria-label="Resources for Continued Learning"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#online-communities class=md-nav__link> <span class=md-ellipsis> Online Communities </span> </a> </li> <li class=md-nav__item> <a href=#software-documentation class=md-nav__link> <span class=md-ellipsis> Software Documentation </span> </a> </li> <li class=md-nav__item> <a href=#professional-development class=md-nav__link> <span class=md-ellipsis> Professional Development </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#looking-ahead class=md-nav__link> <span class=md-ellipsis> Looking Ahead </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../day10/ class=md-nav__link> <span class=md-ellipsis> Day 10 - Wrap-up Session </span> </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--pruned md-nav__item--nested"> <a href=../../datasets/ class=md-nav__link> <span class=md-ellipsis> Resources </span> <span class="md-nav__icon md-icon"></span> </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/CIDRI-Africa/microbial-genomics-training/edit/master/docs/modules/day9/index.md title="Edit this page" class="md-content__button md-icon" rel=edit> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4zm10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1z"/></svg> </a> <a href=https://github.com/CIDRI-Africa/microbial-genomics-training/raw/master/docs/modules/day9/index.md title="View source of this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.2 8.2 0 0 1-1.23-2"/></svg> </a> <h1 id=day-9-bring-your-own-data>Day 9: Bring Your Own Data<a class=headerlink href=#day-9-bring-your-own-data title="Permanent link">&para;</a></h1> <p><strong>Date</strong>: September 11, 2025<br> <strong>Duration</strong>: 09:00-13:00 CAT<br> <strong>Focus</strong>: Independent analysis of participant datasets</p> <h2 id=overview>Overview<a class=headerlink href=#overview title="Permanent link">&para;</a></h2> <p>Day 9 is dedicated to hands-on application of all techniques learned throughout the course. Participants will analyze their own datasets with guidance from trainers, troubleshoot real-world challenges, and develop customized analysis approaches for their specific research questions.</p> <h2 id=learning-objectives>Learning Objectives<a class=headerlink href=#learning-objectives title="Permanent link">&para;</a></h2> <p>By the end of Day 9, you will be able to:</p> <ul> <li>Apply learned bioinformatics techniques to your own research data</li> <li>Troubleshoot common issues encountered in real-world analyses</li> <li>Adapt standard protocols to meet specific research requirements</li> <li>Develop analysis strategies for novel research questions</li> <li>Integrate multiple analysis approaches into comprehensive workflows</li> <li>Plan sustainable bioinformatics practices for ongoing research</li> </ul> <h2 id=schedule>Schedule<a class=headerlink href=#schedule title="Permanent link">&para;</a></h2> <table> <thead> <tr> <th>Time (CAT)</th> <th>Topic</th> <th>Links</th> <th>Trainer</th> </tr> </thead> <tbody> <tr> <td><strong>09:00</strong></td> <td><em>Participants to analyse their own data</em></td> <td><a href=#>Resources</a> • <a href=#>Templates</a></td> <td>All trainers</td> </tr> <tr> <td><strong>11:30</strong></td> <td><strong>Break</strong></td> <td></td> <td></td> </tr> <tr> <td><strong>12:00</strong></td> <td><em>Participants to analyse their own data</em></td> <td><a href=#>Help Desk</a> • <a href=#>Examples</a></td> <td>All trainers</td> </tr> </tbody> </table> <h2 id=session-structure>Session Structure<a class=headerlink href=#session-structure title="Permanent link">&para;</a></h2> <h3 id=opening-session-0900-0915>Opening Session (09:00-09:15)<a class=headerlink href=#opening-session-0900-0915 title="Permanent link">&para;</a></h3> <ul> <li>Brief recap of key course concepts</li> <li>Overview of available support and resources</li> <li>Formation of analysis groups based on data types</li> <li>Technical setup verification</li> </ul> <h3 id=individual-analysis-time-0915-1130>Individual Analysis Time (09:15-11:30)<a class=headerlink href=#individual-analysis-time-0915-1130 title="Permanent link">&para;</a></h3> <ul> <li>Independent work on participant datasets</li> <li>One-on-one consultation with trainers</li> <li>Peer collaboration and knowledge sharing</li> <li>Documentation of analysis approaches</li> </ul> <h3 id=progress-sharing-1145-1200>Progress Sharing (11:45-12:00)<a class=headerlink href=#progress-sharing-1145-1200 title="Permanent link">&para;</a></h3> <ul> <li>Brief presentations of initial findings</li> <li>Discussion of challenges encountered</li> <li>Sharing of successful analysis strategies</li> </ul> <h3 id=advanced-analysis-1200-1300>Advanced Analysis (12:00-13:00)<a class=headerlink href=#advanced-analysis-1200-1300 title="Permanent link">&para;</a></h3> <ul> <li>Continued independent work</li> <li>Focus on complex analyses or troubleshooting</li> <li>Preparation for Day 10 presentations</li> <li>Final consultations with trainers</li> </ul> <h2 id=data-types-and-analysis-approaches>Data Types and Analysis Approaches<a class=headerlink href=#data-types-and-analysis-approaches title="Permanent link">&para;</a></h2> <h3 id=genomic-data>Genomic Data<a class=headerlink href=#genomic-data title="Permanent link">&para;</a></h3> <p><strong>Common analyses for participants with genomic datasets:</strong></p> <ul> <li><strong>Quality control and preprocessing</strong></li> <li>FastQC assessment and interpretation</li> <li>Adapter trimming and quality filtering</li> <li> <p>Species identification and contamination detection</p> </li> <li> <p><strong>Genome assembly and annotation</strong></p> </li> <li>De novo assembly optimization</li> <li>Assembly quality assessment</li> <li> <p>Functional annotation and gene prediction</p> </li> <li> <p><strong>Comparative genomics</strong></p> </li> <li>MLST and serotyping</li> <li>Antimicrobial resistance gene detection</li> <li>Phylogenetic analysis and clustering</li> </ul> <h3 id=metagenomic-data>Metagenomic Data<a class=headerlink href=#metagenomic-data title="Permanent link">&para;</a></h3> <p><strong>For participants with microbiome or metagenomic samples:</strong></p> <ul> <li><strong>Community profiling</strong></li> <li>Taxonomic classification</li> <li>Abundance estimation and normalization</li> <li> <p>Diversity analysis (alpha and beta)</p> </li> <li> <p><strong>Functional analysis</strong></p> </li> <li>Pathway reconstruction</li> <li>Antimicrobial resistance profiling</li> <li> <p>Metabolic potential assessment</p> </li> <li> <p><strong>Clinical applications</strong></p> </li> <li>Pathogen detection in complex samples</li> <li>Co-infection analysis</li> <li>Treatment response monitoring</li> </ul> <h3 id=outbreak-investigation-data>Outbreak Investigation Data<a class=headerlink href=#outbreak-investigation-data title="Permanent link">&para;</a></h3> <p><strong>For epidemiological and outbreak datasets:</strong></p> <ul> <li><strong>Transmission analysis</strong></li> <li>SNP-based clustering</li> <li>Phylogenetic reconstruction</li> <li> <p>Temporal and geographic analysis</p> </li> <li> <p><strong>Resistance surveillance</strong></p> </li> <li>Multi-drug resistance patterns</li> <li>Resistance gene distribution</li> <li>Treatment outcome correlations</li> </ul> <h2 id=technical-support-available>Technical Support Available<a class=headerlink href=#technical-support-available title="Permanent link">&para;</a></h2> <h3 id=computational-resources>Computational Resources<a class=headerlink href=#computational-resources title="Permanent link">&para;</a></h3> <ul> <li>Access to high-performance computing cluster</li> <li>Pre-installed bioinformatics software environments</li> <li>Container images for reproducible analysis</li> <li>Shared storage for large datasets</li> </ul> <h3 id=analysis-pipelines>Analysis Pipelines<a class=headerlink href=#analysis-pipelines title="Permanent link">&para;</a></h3> <ul> <li>Nextflow workflows developed during the course</li> <li>Customizable analysis templates</li> <li>Pre-configured environment profiles</li> <li>Automated reporting tools</li> </ul> <h3 id=expert-guidance>Expert Guidance<a class=headerlink href=#expert-guidance title="Permanent link">&para;</a></h3> <p><strong>Trainer specializations available:</strong></p> <table> <thead> <tr> <th>Trainer</th> <th>Expertise Areas</th> </tr> </thead> <tbody> <tr> <td><strong>Ephifania Geza</strong></td> <td>Genomic surveillance, AMR analysis, metagenomics, clinical applications</td> </tr> <tr> <td><strong>Arash Iranzadeh</strong></td> <td>Phylogenomics, comparative genomics, outbreak investigation</td> </tr> <tr> <td><strong>Sindiswa Lukhele</strong></td> <td>Sequencing technologies, quality control, species identification</td> </tr> <tr> <td><strong>Mamana Mbiyavanga</strong></td> <td>Workflow development, HPC systems, pipeline optimization</td> </tr> </tbody> </table> <h2 id=common-analysis-workflows>Common Analysis Workflows<a class=headerlink href=#common-analysis-workflows title="Permanent link">&para;</a></h2> <h3 id=genomic-surveillance-workflow>Genomic Surveillance Workflow<a class=headerlink href=#genomic-surveillance-workflow title="Permanent link">&para;</a></h3> <div class="language-bash highlight"><pre><span></span><code><span id=__span-0-1><a id=__codelineno-0-1 name=__codelineno-0-1 href=#__codelineno-0-1></a><span class=c1># 1. Initial data assessment</span>
</span><span id=__span-0-2><a id=__codelineno-0-2 name=__codelineno-0-2 href=#__codelineno-0-2></a>fastqc<span class=w> </span>raw_data/*.fastq.gz
</span><span id=__span-0-3><a id=__codelineno-0-3 name=__codelineno-0-3 href=#__codelineno-0-3></a>multiqc<span class=w> </span>fastqc_results/
</span><span id=__span-0-4><a id=__codelineno-0-4 name=__codelineno-0-4 href=#__codelineno-0-4></a>
</span><span id=__span-0-5><a id=__codelineno-0-5 name=__codelineno-0-5 href=#__codelineno-0-5></a><span class=c1># 2. Species identification</span>
</span><span id=__span-0-6><a id=__codelineno-0-6 name=__codelineno-0-6 href=#__codelineno-0-6></a>kraken2<span class=w> </span>--db<span class=w> </span>minikraken2_v2<span class=w> </span>--paired<span class=w> </span>sample_R1.fastq<span class=w> </span>sample_R2.fastq
</span><span id=__span-0-7><a id=__codelineno-0-7 name=__codelineno-0-7 href=#__codelineno-0-7></a>
</span><span id=__span-0-8><a id=__codelineno-0-8 name=__codelineno-0-8 href=#__codelineno-0-8></a><span class=c1># 3. Quality trimming</span>
</span><span id=__span-0-9><a id=__codelineno-0-9 name=__codelineno-0-9 href=#__codelineno-0-9></a>trimmomatic<span class=w> </span>PE<span class=w> </span>sample_R1.fastq<span class=w> </span>sample_R2.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-0-10><a id=__codelineno-0-10 name=__codelineno-0-10 href=#__codelineno-0-10></a><span class=w>    </span>sample_R1_trimmed.fastq<span class=w> </span>sample_R1_unpaired.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-0-11><a id=__codelineno-0-11 name=__codelineno-0-11 href=#__codelineno-0-11></a><span class=w>    </span>sample_R2_trimmed.fastq<span class=w> </span>sample_R2_unpaired.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-0-12><a id=__codelineno-0-12 name=__codelineno-0-12 href=#__codelineno-0-12></a><span class=w>    </span>SLIDINGWINDOW:4:20<span class=w> </span>MINLEN:50
</span><span id=__span-0-13><a id=__codelineno-0-13 name=__codelineno-0-13 href=#__codelineno-0-13></a>
</span><span id=__span-0-14><a id=__codelineno-0-14 name=__codelineno-0-14 href=#__codelineno-0-14></a><span class=c1># 4. Assembly</span>
</span><span id=__span-0-15><a id=__codelineno-0-15 name=__codelineno-0-15 href=#__codelineno-0-15></a>spades.py<span class=w> </span>-1<span class=w> </span>sample_R1_trimmed.fastq<span class=w> </span>-2<span class=w> </span>sample_R2_trimmed.fastq<span class=w> </span>-o<span class=w> </span>assembly/
</span><span id=__span-0-16><a id=__codelineno-0-16 name=__codelineno-0-16 href=#__codelineno-0-16></a>
</span><span id=__span-0-17><a id=__codelineno-0-17 name=__codelineno-0-17 href=#__codelineno-0-17></a><span class=c1># 5. Assembly quality assessment</span>
</span><span id=__span-0-18><a id=__codelineno-0-18 name=__codelineno-0-18 href=#__codelineno-0-18></a>quast.py<span class=w> </span>assembly/scaffolds.fasta<span class=w> </span>-o<span class=w> </span>quast_results/
</span><span id=__span-0-19><a id=__codelineno-0-19 name=__codelineno-0-19 href=#__codelineno-0-19></a>
</span><span id=__span-0-20><a id=__codelineno-0-20 name=__codelineno-0-20 href=#__codelineno-0-20></a><span class=c1># 6. Annotation</span>
</span><span id=__span-0-21><a id=__codelineno-0-21 name=__codelineno-0-21 href=#__codelineno-0-21></a>prokka<span class=w> </span>assembly/scaffolds.fasta<span class=w> </span>--outdir<span class=w> </span>annotation/<span class=w> </span>--prefix<span class=w> </span>sample
</span></code></pre></div> <h3 id=metagenomic-analysis-workflow>Metagenomic Analysis Workflow<a class=headerlink href=#metagenomic-analysis-workflow title="Permanent link">&para;</a></h3> <div class="language-bash highlight"><pre><span></span><code><span id=__span-1-1><a id=__codelineno-1-1 name=__codelineno-1-1 href=#__codelineno-1-1></a><span class=c1># 1. Host DNA removal (if applicable)</span>
</span><span id=__span-1-2><a id=__codelineno-1-2 name=__codelineno-1-2 href=#__codelineno-1-2></a>kneaddata<span class=w> </span>--input<span class=w> </span>sample_R1.fastq<span class=w> </span>--input<span class=w> </span>sample_R2.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-1-3><a id=__codelineno-1-3 name=__codelineno-1-3 href=#__codelineno-1-3></a><span class=w>    </span>--reference-db<span class=w> </span>human_genome<span class=w> </span>--output<span class=w> </span>cleaned_data/
</span><span id=__span-1-4><a id=__codelineno-1-4 name=__codelineno-1-4 href=#__codelineno-1-4></a>
</span><span id=__span-1-5><a id=__codelineno-1-5 name=__codelineno-1-5 href=#__codelineno-1-5></a><span class=c1># 2. Taxonomic profiling</span>
</span><span id=__span-1-6><a id=__codelineno-1-6 name=__codelineno-1-6 href=#__codelineno-1-6></a>metaphlan<span class=w> </span>cleaned_data/sample_paired_1.fastq,cleaned_data/sample_paired_2.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-1-7><a id=__codelineno-1-7 name=__codelineno-1-7 href=#__codelineno-1-7></a><span class=w>    </span>--bowtie2out<span class=w> </span>sample.bowtie2.bz2<span class=w> </span>--nproc<span class=w> </span><span class=m>4</span><span class=w> </span>--input_type<span class=w> </span>fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-1-8><a id=__codelineno-1-8 name=__codelineno-1-8 href=#__codelineno-1-8></a><span class=w>    </span>--output_file<span class=w> </span>sample_profile.txt
</span><span id=__span-1-9><a id=__codelineno-1-9 name=__codelineno-1-9 href=#__codelineno-1-9></a>
</span><span id=__span-1-10><a id=__codelineno-1-10 name=__codelineno-1-10 href=#__codelineno-1-10></a><span class=c1># 3. Functional profiling</span>
</span><span id=__span-1-11><a id=__codelineno-1-11 name=__codelineno-1-11 href=#__codelineno-1-11></a>humann<span class=w> </span>--input<span class=w> </span>cleaned_data/sample_paired.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-1-12><a id=__codelineno-1-12 name=__codelineno-1-12 href=#__codelineno-1-12></a><span class=w>    </span>--output<span class=w> </span>functional_analysis/<span class=w> </span>--nucleotide-database<span class=w> </span>chocophlan<span class=w> </span><span class=se>\</span>
</span><span id=__span-1-13><a id=__codelineno-1-13 name=__codelineno-1-13 href=#__codelineno-1-13></a><span class=w>    </span>--protein-database<span class=w> </span>uniref90
</span><span id=__span-1-14><a id=__codelineno-1-14 name=__codelineno-1-14 href=#__codelineno-1-14></a>
</span><span id=__span-1-15><a id=__codelineno-1-15 name=__codelineno-1-15 href=#__codelineno-1-15></a><span class=c1># 4. Diversity analysis in R</span>
</span><span id=__span-1-16><a id=__codelineno-1-16 name=__codelineno-1-16 href=#__codelineno-1-16></a>Rscript<span class=w> </span>diversity_analysis.R<span class=w> </span>sample_profile.txt<span class=w> </span>metadata.csv
</span></code></pre></div> <h2 id=troubleshooting-guide>Troubleshooting Guide<a class=headerlink href=#troubleshooting-guide title="Permanent link">&para;</a></h2> <h3 id=common-issues-and-solutions>Common Issues and Solutions<a class=headerlink href=#common-issues-and-solutions title="Permanent link">&para;</a></h3> <h4 id=low-quality-data>Low-Quality Data<a class=headerlink href=#low-quality-data title="Permanent link">&para;</a></h4> <div class="language-bash highlight"><pre><span></span><code><span id=__span-2-1><a id=__codelineno-2-1 name=__codelineno-2-1 href=#__codelineno-2-1></a><span class=c1># Check read quality distribution</span>
</span><span id=__span-2-2><a id=__codelineno-2-2 name=__codelineno-2-2 href=#__codelineno-2-2></a>fastqc<span class=w> </span>*.fastq.gz
</span><span id=__span-2-3><a id=__codelineno-2-3 name=__codelineno-2-3 href=#__codelineno-2-3></a>
</span><span id=__span-2-4><a id=__codelineno-2-4 name=__codelineno-2-4 href=#__codelineno-2-4></a><span class=c1># Aggressive quality trimming if needed</span>
</span><span id=__span-2-5><a id=__codelineno-2-5 name=__codelineno-2-5 href=#__codelineno-2-5></a>trimmomatic<span class=w> </span>PE<span class=w> </span>input_R1.fastq<span class=w> </span>input_R2.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-2-6><a id=__codelineno-2-6 name=__codelineno-2-6 href=#__codelineno-2-6></a><span class=w>    </span>output_R1.fastq<span class=w> </span>unpaired_R1.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-2-7><a id=__codelineno-2-7 name=__codelineno-2-7 href=#__codelineno-2-7></a><span class=w>    </span>output_R2.fastq<span class=w> </span>unpaired_R2.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-2-8><a id=__codelineno-2-8 name=__codelineno-2-8 href=#__codelineno-2-8></a><span class=w>    </span>SLIDINGWINDOW:4:25<span class=w> </span>LEADING:20<span class=w> </span>TRAILING:20<span class=w> </span>MINLEN:75
</span><span id=__span-2-9><a id=__codelineno-2-9 name=__codelineno-2-9 href=#__codelineno-2-9></a>
</span><span id=__span-2-10><a id=__codelineno-2-10 name=__codelineno-2-10 href=#__codelineno-2-10></a><span class=c1># Consider different assembly strategies</span>
</span><span id=__span-2-11><a id=__codelineno-2-11 name=__codelineno-2-11 href=#__codelineno-2-11></a><span class=c1># For poor quality data, try more conservative parameters</span>
</span><span id=__span-2-12><a id=__codelineno-2-12 name=__codelineno-2-12 href=#__codelineno-2-12></a>spades.py<span class=w> </span>--careful<span class=w> </span>--cov-cutoff<span class=w> </span>auto<span class=w> </span>-1<span class=w> </span>R1.fastq<span class=w> </span>-2<span class=w> </span>R2.fastq<span class=w> </span>-o<span class=w> </span>assembly/
</span></code></pre></div> <h4 id=contamination-issues>Contamination Issues<a class=headerlink href=#contamination-issues title="Permanent link">&para;</a></h4> <div class="language-bash highlight"><pre><span></span><code><span id=__span-3-1><a id=__codelineno-3-1 name=__codelineno-3-1 href=#__codelineno-3-1></a><span class=c1># Check for contamination</span>
</span><span id=__span-3-2><a id=__codelineno-3-2 name=__codelineno-3-2 href=#__codelineno-3-2></a>kraken2<span class=w> </span>--db<span class=w> </span>standard<span class=w> </span>--paired<span class=w> </span>sample_R1.fastq<span class=w> </span>sample_R2.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-3-3><a id=__codelineno-3-3 name=__codelineno-3-3 href=#__codelineno-3-3></a><span class=w>    </span>--report<span class=w> </span>contamination_report.txt
</span><span id=__span-3-4><a id=__codelineno-3-4 name=__codelineno-3-4 href=#__codelineno-3-4></a>
</span><span id=__span-3-5><a id=__codelineno-3-5 name=__codelineno-3-5 href=#__codelineno-3-5></a><span class=c1># Remove contaminant sequences</span>
</span><span id=__span-3-6><a id=__codelineno-3-6 name=__codelineno-3-6 href=#__codelineno-3-6></a>extract_kraken_reads.py<span class=w> </span>-k<span class=w> </span>sample.kraken<span class=w> </span>-s1<span class=w> </span>sample_R1.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-3-7><a id=__codelineno-3-7 name=__codelineno-3-7 href=#__codelineno-3-7></a><span class=w>    </span>-s2<span class=w> </span>sample_R2.fastq<span class=w> </span>-o<span class=w> </span>clean_R1.fastq<span class=w> </span>-o2<span class=w> </span>clean_R2.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-3-8><a id=__codelineno-3-8 name=__codelineno-3-8 href=#__codelineno-3-8></a><span class=w>    </span>--exclude<span class=w> </span>--taxid<span class=w> </span><span class=m>9606</span><span class=w>  </span><span class=c1># Exclude human reads</span>
</span></code></pre></div> <h4 id=assembly-problems>Assembly Problems<a class=headerlink href=#assembly-problems title="Permanent link">&para;</a></h4> <div class="language-bash highlight"><pre><span></span><code><span id=__span-4-1><a id=__codelineno-4-1 name=__codelineno-4-1 href=#__codelineno-4-1></a><span class=c1># If SPAdes fails, try different assemblers</span>
</span><span id=__span-4-2><a id=__codelineno-4-2 name=__codelineno-4-2 href=#__codelineno-4-2></a><span class=c1># Unicycler for hybrid assembly</span>
</span><span id=__span-4-3><a id=__codelineno-4-3 name=__codelineno-4-3 href=#__codelineno-4-3></a>unicycler<span class=w> </span>-1<span class=w> </span>short_R1.fastq<span class=w> </span>-2<span class=w> </span>short_R2.fastq<span class=w> </span>-l<span class=w> </span>long_reads.fastq<span class=w> </span>-o<span class=w> </span>assembly/
</span><span id=__span-4-4><a id=__codelineno-4-4 name=__codelineno-4-4 href=#__codelineno-4-4></a>
</span><span id=__span-4-5><a id=__codelineno-4-5 name=__codelineno-4-5 href=#__codelineno-4-5></a><span class=c1># Or SKESA for quick assembly</span>
</span><span id=__span-4-6><a id=__codelineno-4-6 name=__codelineno-4-6 href=#__codelineno-4-6></a>skesa<span class=w> </span>--reads<span class=w> </span>short_R1.fastq,short_R2.fastq<span class=w> </span>--cores<span class=w> </span><span class=m>8</span><span class=w> </span>&gt;<span class=w> </span>assembly.fasta
</span></code></pre></div> <h4 id=memoryresource-issues>Memory/Resource Issues<a class=headerlink href=#memoryresource-issues title="Permanent link">&para;</a></h4> <div class="language-bash highlight"><pre><span></span><code><span id=__span-5-1><a id=__codelineno-5-1 name=__codelineno-5-1 href=#__codelineno-5-1></a><span class=c1># Monitor resource usage</span>
</span><span id=__span-5-2><a id=__codelineno-5-2 name=__codelineno-5-2 href=#__codelineno-5-2></a>htop
</span><span id=__span-5-3><a id=__codelineno-5-3 name=__codelineno-5-3 href=#__codelineno-5-3></a>
</span><span id=__span-5-4><a id=__codelineno-5-4 name=__codelineno-5-4 href=#__codelineno-5-4></a><span class=c1># Reduce memory usage for large datasets</span>
</span><span id=__span-5-5><a id=__codelineno-5-5 name=__codelineno-5-5 href=#__codelineno-5-5></a>spades.py<span class=w> </span>--memory<span class=w> </span><span class=m>16</span><span class=w> </span>-1<span class=w> </span>R1.fastq<span class=w> </span>-2<span class=w> </span>R2.fastq<span class=w> </span>-o<span class=w> </span>assembly/
</span><span id=__span-5-6><a id=__codelineno-5-6 name=__codelineno-5-6 href=#__codelineno-5-6></a>
</span><span id=__span-5-7><a id=__codelineno-5-7 name=__codelineno-5-7 href=#__codelineno-5-7></a><span class=c1># Use subsampling for initial testing</span>
</span><span id=__span-5-8><a id=__codelineno-5-8 name=__codelineno-5-8 href=#__codelineno-5-8></a>seqtk<span class=w> </span>sample<span class=w> </span>-s100<span class=w> </span>input_R1.fastq<span class=w> </span><span class=m>100000</span><span class=w> </span>&gt;<span class=w> </span>subset_R1.fastq
</span><span id=__span-5-9><a id=__codelineno-5-9 name=__codelineno-5-9 href=#__codelineno-5-9></a>seqtk<span class=w> </span>sample<span class=w> </span>-s100<span class=w> </span>input_R2.fastq<span class=w> </span><span class=m>100000</span><span class=w> </span>&gt;<span class=w> </span>subset_R2.fastq
</span></code></pre></div> <h2 id=analysis-documentation>Analysis Documentation<a class=headerlink href=#analysis-documentation title="Permanent link">&para;</a></h2> <h3 id=laboratory-notebook-template>Laboratory Notebook Template<a class=headerlink href=#laboratory-notebook-template title="Permanent link">&para;</a></h3> <div class="language-markdown highlight"><pre><span></span><code><span id=__span-6-1><a id=__codelineno-6-1 name=__codelineno-6-1 href=#__codelineno-6-1></a><span class=gh># Analysis Log: [Your Dataset Name]</span>
</span><span id=__span-6-2><a id=__codelineno-6-2 name=__codelineno-6-2 href=#__codelineno-6-2></a><span class=gs>**Date**</span>: [Current Date]
</span><span id=__span-6-3><a id=__codelineno-6-3 name=__codelineno-6-3 href=#__codelineno-6-3></a><span class=gs>**Analyst**</span>: [Your Name]
</span><span id=__span-6-4><a id=__codelineno-6-4 name=__codelineno-6-4 href=#__codelineno-6-4></a><span class=gs>**Data Source**</span>: [Description of samples]
</span><span id=__span-6-5><a id=__codelineno-6-5 name=__codelineno-6-5 href=#__codelineno-6-5></a>
</span><span id=__span-6-6><a id=__codelineno-6-6 name=__codelineno-6-6 href=#__codelineno-6-6></a><span class=gu>## Objectives</span>
</span><span id=__span-6-7><a id=__codelineno-6-7 name=__codelineno-6-7 href=#__codelineno-6-7></a><span class=k>-</span><span class=w> </span>Primary research question
</span><span id=__span-6-8><a id=__codelineno-6-8 name=__codelineno-6-8 href=#__codelineno-6-8></a><span class=k>-</span><span class=w> </span>Specific analyses planned
</span><span id=__span-6-9><a id=__codelineno-6-9 name=__codelineno-6-9 href=#__codelineno-6-9></a><span class=k>-</span><span class=w> </span>Expected outcomes
</span><span id=__span-6-10><a id=__codelineno-6-10 name=__codelineno-6-10 href=#__codelineno-6-10></a>
</span><span id=__span-6-11><a id=__codelineno-6-11 name=__codelineno-6-11 href=#__codelineno-6-11></a><span class=gu>## Data Description</span>
</span><span id=__span-6-12><a id=__codelineno-6-12 name=__codelineno-6-12 href=#__codelineno-6-12></a><span class=k>-</span><span class=w> </span>Sample type and collection method
</span><span id=__span-6-13><a id=__codelineno-6-13 name=__codelineno-6-13 href=#__codelineno-6-13></a><span class=k>-</span><span class=w> </span>Sequencing platform and parameters
</span><span id=__span-6-14><a id=__codelineno-6-14 name=__codelineno-6-14 href=#__codelineno-6-14></a><span class=k>-</span><span class=w> </span>Data quality metrics
</span><span id=__span-6-15><a id=__codelineno-6-15 name=__codelineno-6-15 href=#__codelineno-6-15></a>
</span><span id=__span-6-16><a id=__codelineno-6-16 name=__codelineno-6-16 href=#__codelineno-6-16></a><span class=gu>## Analysis Steps</span>
</span><span id=__span-6-17><a id=__codelineno-6-17 name=__codelineno-6-17 href=#__codelineno-6-17></a><span class=gu>### Step 1: Quality Control</span>
</span><span id=__span-6-18><a id=__codelineno-6-18 name=__codelineno-6-18 href=#__codelineno-6-18></a><span class=k>-</span><span class=w> </span>Command used: <span class=sb>`fastqc *.fastq.gz`</span>
</span><span id=__span-6-19><a id=__codelineno-6-19 name=__codelineno-6-19 href=#__codelineno-6-19></a><span class=k>-</span><span class=w> </span>Results: [Summary of quality metrics]
</span><span id=__span-6-20><a id=__codelineno-6-20 name=__codelineno-6-20 href=#__codelineno-6-20></a><span class=k>-</span><span class=w> </span>Decision: [Any quality filtering applied]
</span><span id=__span-6-21><a id=__codelineno-6-21 name=__codelineno-6-21 href=#__codelineno-6-21></a>
</span><span id=__span-6-22><a id=__codelineno-6-22 name=__codelineno-6-22 href=#__codelineno-6-22></a><span class=gu>### Step 2: [Next Analysis]</span>
</span><span id=__span-6-23><a id=__codelineno-6-23 name=__codelineno-6-23 href=#__codelineno-6-23></a><span class=k>-</span><span class=w> </span>Command: [Exact command used]
</span><span id=__span-6-24><a id=__codelineno-6-24 name=__codelineno-6-24 href=#__codelineno-6-24></a><span class=k>-</span><span class=w> </span>Parameters chosen: [Rationale for parameter selection]
</span><span id=__span-6-25><a id=__codelineno-6-25 name=__codelineno-6-25 href=#__codelineno-6-25></a><span class=k>-</span><span class=w> </span>Results: [Key findings]
</span><span id=__span-6-26><a id=__codelineno-6-26 name=__codelineno-6-26 href=#__codelineno-6-26></a>
</span><span id=__span-6-27><a id=__codelineno-6-27 name=__codelineno-6-27 href=#__codelineno-6-27></a><span class=gu>## Challenges Encountered</span>
</span><span id=__span-6-28><a id=__codelineno-6-28 name=__codelineno-6-28 href=#__codelineno-6-28></a><span class=k>-</span><span class=w> </span>Issue: [Description of problem]
</span><span id=__span-6-29><a id=__codelineno-6-29 name=__codelineno-6-29 href=#__codelineno-6-29></a><span class=k>-</span><span class=w> </span>Solution attempted: [What was tried]
</span><span id=__span-6-30><a id=__codelineno-6-30 name=__codelineno-6-30 href=#__codelineno-6-30></a><span class=k>-</span><span class=w> </span>Outcome: [Whether resolved]
</span><span id=__span-6-31><a id=__codelineno-6-31 name=__codelineno-6-31 href=#__codelineno-6-31></a>
</span><span id=__span-6-32><a id=__codelineno-6-32 name=__codelineno-6-32 href=#__codelineno-6-32></a><span class=gu>## Key Findings</span>
</span><span id=__span-6-33><a id=__codelineno-6-33 name=__codelineno-6-33 href=#__codelineno-6-33></a><span class=k>-</span><span class=w> </span>[Major results from analysis]
</span><span id=__span-6-34><a id=__codelineno-6-34 name=__codelineno-6-34 href=#__codelineno-6-34></a><span class=k>-</span><span class=w> </span>[Statistical summaries]
</span><span id=__span-6-35><a id=__codelineno-6-35 name=__codelineno-6-35 href=#__codelineno-6-35></a><span class=k>-</span><span class=w> </span>[Biological interpretations]
</span><span id=__span-6-36><a id=__codelineno-6-36 name=__codelineno-6-36 href=#__codelineno-6-36></a>
</span><span id=__span-6-37><a id=__codelineno-6-37 name=__codelineno-6-37 href=#__codelineno-6-37></a><span class=gu>## Next Steps</span>
</span><span id=__span-6-38><a id=__codelineno-6-38 name=__codelineno-6-38 href=#__codelineno-6-38></a><span class=k>-</span><span class=w> </span>Additional analyses needed
</span><span id=__span-6-39><a id=__codelineno-6-39 name=__codelineno-6-39 href=#__codelineno-6-39></a><span class=k>-</span><span class=w> </span>Questions raised
</span><span id=__span-6-40><a id=__codelineno-6-40 name=__codelineno-6-40 href=#__codelineno-6-40></a><span class=k>-</span><span class=w> </span>Follow-up experiments
</span></code></pre></div> <h2 id=resource-management>Resource Management<a class=headerlink href=#resource-management title="Permanent link">&para;</a></h2> <h3 id=data-organization>Data Organization<a class=headerlink href=#data-organization title="Permanent link">&para;</a></h3> <div class="language-bash highlight"><pre><span></span><code><span id=__span-7-1><a id=__codelineno-7-1 name=__codelineno-7-1 href=#__codelineno-7-1></a><span class=c1># Recommended directory structure</span>
</span><span id=__span-7-2><a id=__codelineno-7-2 name=__codelineno-7-2 href=#__codelineno-7-2></a>project_name/
</span><span id=__span-7-3><a id=__codelineno-7-3 name=__codelineno-7-3 href=#__codelineno-7-3></a>├──<span class=w> </span>raw_data/<span class=w>          </span><span class=c1># Original sequencing files</span>
</span><span id=__span-7-4><a id=__codelineno-7-4 name=__codelineno-7-4 href=#__codelineno-7-4></a>├──<span class=w> </span>quality_control/<span class=w>   </span><span class=c1># QC reports and cleaned data</span>
</span><span id=__span-7-5><a id=__codelineno-7-5 name=__codelineno-7-5 href=#__codelineno-7-5></a>├──<span class=w> </span>analysis/<span class=w>          </span><span class=c1># Main analysis outputs</span>
</span><span id=__span-7-6><a id=__codelineno-7-6 name=__codelineno-7-6 href=#__codelineno-7-6></a>├──<span class=w> </span>scripts/<span class=w>           </span><span class=c1># Custom scripts and commands</span>
</span><span id=__span-7-7><a id=__codelineno-7-7 name=__codelineno-7-7 href=#__codelineno-7-7></a>├──<span class=w> </span>results/<span class=w>           </span><span class=c1># Final results and figures</span>
</span><span id=__span-7-8><a id=__codelineno-7-8 name=__codelineno-7-8 href=#__codelineno-7-8></a>└──<span class=w> </span>documentation/<span class=w>     </span><span class=c1># Analysis logs and notes</span>
</span></code></pre></div> <h3 id=backup-strategies>Backup Strategies<a class=headerlink href=#backup-strategies title="Permanent link">&para;</a></h3> <div class="language-bash highlight"><pre><span></span><code><span id=__span-8-1><a id=__codelineno-8-1 name=__codelineno-8-1 href=#__codelineno-8-1></a><span class=c1># Regular backup of important results</span>
</span><span id=__span-8-2><a id=__codelineno-8-2 name=__codelineno-8-2 href=#__codelineno-8-2></a>rsync<span class=w> </span>-av<span class=w> </span>results/<span class=w> </span>backup_drive/project_results/
</span><span id=__span-8-3><a id=__codelineno-8-3 name=__codelineno-8-3 href=#__codelineno-8-3></a>tar<span class=w> </span>-czf<span class=w> </span>analysis_<span class=k>$(</span>date<span class=w> </span>+%Y%m%d<span class=k>)</span>.tar.gz<span class=w> </span>analysis/
</span><span id=__span-8-4><a id=__codelineno-8-4 name=__codelineno-8-4 href=#__codelineno-8-4></a>
</span><span id=__span-8-5><a id=__codelineno-8-5 name=__codelineno-8-5 href=#__codelineno-8-5></a><span class=c1># Version control for scripts</span>
</span><span id=__span-8-6><a id=__codelineno-8-6 name=__codelineno-8-6 href=#__codelineno-8-6></a>git<span class=w> </span>init
</span><span id=__span-8-7><a id=__codelineno-8-7 name=__codelineno-8-7 href=#__codelineno-8-7></a>git<span class=w> </span>add<span class=w> </span>scripts/
</span><span id=__span-8-8><a id=__codelineno-8-8 name=__codelineno-8-8 href=#__codelineno-8-8></a>git<span class=w> </span>commit<span class=w> </span>-m<span class=w> </span><span class=s2>&quot;Initial analysis scripts&quot;</span>
</span></code></pre></div> <h2 id=collaboration-guidelines>Collaboration Guidelines<a class=headerlink href=#collaboration-guidelines title="Permanent link">&para;</a></h2> <h3 id=peer-support>Peer Support<a class=headerlink href=#peer-support title="Permanent link">&para;</a></h3> <ul> <li>Form analysis groups based on similar data types</li> <li>Share successful parameter combinations</li> <li>Collaborate on troubleshooting challenging datasets</li> <li>Review each other's analysis approaches</li> </ul> <h3 id=trainer-consultation>Trainer Consultation<a class=headerlink href=#trainer-consultation title="Permanent link">&para;</a></h3> <ul> <li>Prepare specific questions about your data</li> <li>Document issues with exact error messages</li> <li>Have your analysis objectives clearly defined</li> <li>Be ready to explain your research context</li> </ul> <h2 id=assessment-and-preparation-for-day-10>Assessment and Preparation for Day 10<a class=headerlink href=#assessment-and-preparation-for-day-10 title="Permanent link">&para;</a></h2> <h3 id=presentation-preparation>Presentation Preparation<a class=headerlink href=#presentation-preparation title="Permanent link">&para;</a></h3> <p>Participants should prepare a 5-minute presentation covering:</p> <ol> <li><strong>Research Question</strong>: What you aimed to investigate</li> <li><strong>Data Overview</strong>: Type and source of your dataset</li> <li><strong>Methods Applied</strong>: Which course techniques you used</li> <li><strong>Key Results</strong>: Main findings from your analysis</li> <li><strong>Challenges</strong>: Obstacles encountered and solutions found</li> <li><strong>Future Directions</strong>: Next steps for your research</li> </ol> <h3 id=technical-documentation>Technical Documentation<a class=headerlink href=#technical-documentation title="Permanent link">&para;</a></h3> <ul> <li>Save all commands used in a script file</li> <li>Document parameter choices and rationale</li> <li>Prepare summary statistics and key figures</li> <li>Note any analysis limitations or assumptions</li> </ul> <h2 id=success-metrics>Success Metrics<a class=headerlink href=#success-metrics title="Permanent link">&para;</a></h2> <p>By the end of Day 9, participants should have:</p> <ul class=task-list> <li class=task-list-item><label class=task-list-control><input type=checkbox disabled><span class=task-list-indicator></span></label> Successfully processed their own dataset</li> <li class=task-list-item><label class=task-list-control><input type=checkbox disabled><span class=task-list-indicator></span></label> Applied at least 3 different analysis techniques from the course</li> <li class=task-list-item><label class=task-list-control><input type=checkbox disabled><span class=task-list-indicator></span></label> Documented their analysis workflow</li> <li class=task-list-item><label class=task-list-control><input type=checkbox disabled><span class=task-list-indicator></span></label> Identified key findings relevant to their research</li> <li class=task-list-item><label class=task-list-control><input type=checkbox disabled><span class=task-list-indicator></span></label> Prepared materials for Day 10 presentation</li> <li class=task-list-item><label class=task-list-control><input type=checkbox disabled><span class=task-list-indicator></span></label> Established ongoing analysis plan</li> </ul> <h2 id=resources-for-continued-learning>Resources for Continued Learning<a class=headerlink href=#resources-for-continued-learning title="Permanent link">&para;</a></h2> <h3 id=online-communities>Online Communities<a class=headerlink href=#online-communities title="Permanent link">&para;</a></h3> <ul> <li><a href=https://bioinformatics.stackexchange.com/ >Bioinformatics Stack Exchange</a></li> <li><a href=https://www.biostars.org/ >BioStars Forum</a></li> <li><a href=https://galaxyproject.org/community/ >Galaxy Community</a></li> </ul> <h3 id=software-documentation>Software Documentation<a class=headerlink href=#software-documentation title="Permanent link">&para;</a></h3> <ul> <li>Tool-specific manuals and tutorials</li> <li>GitHub repositories for pipeline development</li> <li>Container registries for reproducible environments</li> </ul> <h3 id=professional-development>Professional Development<a class=headerlink href=#professional-development title="Permanent link">&para;</a></h3> <ul> <li>Local bioinformatics user groups</li> <li>International conferences and workshops</li> <li>Online course platforms for advanced topics</li> </ul> <h2 id=looking-ahead>Looking Ahead<a class=headerlink href=#looking-ahead title="Permanent link">&para;</a></h2> <p><strong>Day 10 Preview</strong>: Wrap-up session including: - Participant presentations of analysis results - Discussion of lessons learned and best practices - Information about ongoing support resources - Course completion and next steps planning</p> <hr> <p><strong>Key Learning Outcome</strong>: Independent application of bioinformatics skills to real research data builds confidence and reveals the practical challenges and rewards of computational biology in actual research contexts.</p> </article> </div> <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script> <script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script> </div> <button type=button class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg> Back to top </button> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../day8/ class="md-footer__link md-footer__link--prev" aria-label="Previous: Day 8 - Pipeline Development"> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg> </div> <div class=md-footer__title> <span class=md-footer__direction> Previous </span> <div class=md-ellipsis> Day 8 - Pipeline Development </div> </div> </a> <a href=../day10/ class="md-footer__link md-footer__link--next" aria-label="Next: Day 10 - Wrap-up Session"> <div class=md-footer__title> <span class=md-footer__direction> Next </span> <div class=md-ellipsis> Day 10 - Wrap-up Session </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; 2024 CIDRI-Africa </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=https://github.com/CIDRI-Africa target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill=currentColor d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg> </a> <a href=https://twitter.com/CIDRIAfrica target=_blank rel=noopener title=twitter.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill=currentColor d="M459.4 151.7c.3 4.5.3 9.1.3 13.6 0 138.7-105.6 298.6-298.6 298.6-59.5 0-114.7-17.2-161.1-47.1 8.4 1 16.6 1.3 25.3 1.3 49.1 0 94.2-16.6 130.3-44.8-46.1-1-84.8-31.2-98.1-72.8 6.5 1 13 1.6 19.8 1.6 9.4 0 18.8-1.3 27.6-3.6-48.1-9.7-84.1-52-84.1-103v-1.3c14 7.8 30.2 12.7 47.4 13.3-28.3-18.8-46.8-51-46.8-87.4 0-19.5 5.2-37.4 14.3-53C87.4 130.8 165 172.4 252.1 176.9c-1.6-7.8-2.6-15.9-2.6-24C249.5 95.1 296.3 48 354.4 48c30.2 0 57.5 12.7 76.7 33.1 23.7-4.5 46.5-13.3 66.6-25.3-7.8 24.4-24.4 44.8-46.1 57.8 21.1-2.3 41.6-8.1 60.4-16.2-14.3 20.8-32.2 39.3-52.6 54.3"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <div class=md-progress data-md-component=progress role=progressbar></div> <script id=__config type=application/json>{"base": "../..", "features": ["announce.dismiss", "content.action.edit", "content.action.view", "content.code.annotate", "content.code.copy", "content.code.select", "content.tabs.link", "content.tooltips", "header.autohide", "navigation.expand", "navigation.footer", "navigation.indexes", "navigation.instant", "navigation.instant.prefetch", "navigation.instant.progress", "navigation.prune", "navigation.sections", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow", "toc.integrate"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script> <script src=../../assets/javascripts/bundle.92b07e13.min.js></script> <script src=../../javascripts/extra.js></script> <script src=../../javascripts/mathjax.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body> </html>