<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="Hands-on training course for microbial genomics and metagenomics in clinical and public health applications"><meta name=author content=CIDRI-Africa><link href=https://cidri-africa.github.io/microbial-genomics-training/modules/day6/ rel=canonical><link href=../day5/ rel=prev><link href=../day7/ rel=next><link rel=icon href=../../assets/favicon.png><meta name=generator content="mkdocs-1.6.1, mkdocs-material-9.6.18"><title>Day 6 - Nextflow Pipeline Development - Microbial Genomics & Metagenomics Training Course</title><link rel=stylesheet href=../../assets/stylesheets/main.7e37652d.min.css><link rel=stylesheet href=../../assets/stylesheets/palette.06af60db.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><link rel=stylesheet href=../../stylesheets/extra.css><script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script><script id=__analytics>function __md_analytics(){function e(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],e("js",new Date),e("config",""),document.addEventListener("DOMContentLoaded",(function(){document.forms.search&&document.forms.search.query.addEventListener("blur",(function(){this.value&&e("event","search",{search_term:this.value})}));document$.subscribe((function(){var t=document.forms.feedback;if(void 0!==t)for(var a of t.querySelectorAll("[type=submit]"))a.addEventListener("click",(function(a){a.preventDefault();var n=document.location.pathname,d=this.getAttribute("data-md-value");e("event","feedback",{page:n,data:d}),t.firstElementChild.disabled=!0;var r=t.querySelector(".md-feedback__note [data-md-value='"+d+"']");r&&(r.hidden=!1)})),t.hidden=!1})),location$.subscribe((function(t){e("config","",{page_path:t.pathname})}))}));var t=document.createElement("script");t.async=!0,t.src="https://www.googletagmanager.com/gtag/js?id=",document.getElementById("__analytics").insertAdjacentElement("afterEnd",t)}</script><script>"undefined"!=typeof __md_analytics&&__md_analytics()</script> <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }
    .gscrollbar-fixer { padding-right: 15px; }
    .gdesc-inner { font-size: 0.75rem; }
    body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
    body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
    body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}</style> <script src="../../assets/javascripts/glightbox.min.js"></script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary=teal data-md-color-accent=amber> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#day-6-nextflow-pipeline-development class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class="md-header md-header--shadow md-header--lifted" data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../.. title="Microbial Genomics & Metagenomics Training Course" class="md-header__button md-logo" aria-label="Microbial Genomics & Metagenomics Training Course" data-md-component=logo> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 2h2v2c0 1.44.68 2.61 1.88 3.78.86.83 2.01 1.63 3.21 2.42l-1.83 1.19C8.27 10.72 7.31 10 6.5 9.21 5.07 7.82 4 6.1 4 4zm14 0h2v2c0 2.1-1.07 3.82-2.5 5.21-1.41 1.38-3.21 2.52-4.96 3.63-1.75 1.12-3.45 2.21-4.66 3.38C6.68 17.39 6 18.56 6 20v2H4v-2c0-2.1 1.07-3.82 2.5-5.21 1.41-1.38 3.21-2.52 4.96-3.63 1.75-1.12 3.45-2.21 4.66-3.38C17.32 6.61 18 5.44 18 4zm-3.26 10.61c.99.67 1.95 1.39 2.76 2.18C18.93 16.18 20 17.9 20 20v2h-2v-2c0-1.44-.68-2.61-1.88-3.78-.86-.83-2.01-1.63-3.21-2.42zM7 3h10v1l-.06.5H7.06L7 4zm.68 3h8.64c-.24.34-.52.69-.9 1.06l-.51.44H9.07l-.49-.44c-.38-.37-.66-.72-.9-1.06m1.41 10.5h5.84l.49.44c.38.37.66.72.9 1.06H7.68c.24-.34.52-.69.9-1.06zm-2.03 3h9.88l.06.5v1H7v-1z"/></svg> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Microbial Genomics & Metagenomics Training Course </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Day 6 - Nextflow Pipeline Development </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media data-md-color-scheme=default data-md-color-primary=teal data-md-color-accent=amber aria-label="Switch to dark mode" type=radio name=__palette id=__palette_0> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg> </label> <input class=md-option data-md-color-media data-md-color-scheme=slate data-md-color-primary=teal data-md-color-accent=amber aria-label="Switch to light mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_0 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg> </label> </form> <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg> </label> <nav class=md-search__options aria-label=Search> <a href=javascript:void(0) class="md-search__icon md-icon" title=Share aria-label=Share data-clipboard data-clipboard-text data-md-component=search-share tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg> </a> <button type=reset class="md-search__icon md-icon" title=Clear aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg> </button> </nav> <div class=md-search__suggest data-md-component=search-suggest></div> </form> <div class=md-search__output> <div class=md-search__scrollwrap tabindex=0 data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list role=presentation></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/CIDRI-Africa/microbial-genomics-training title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill=currentColor d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg> </div> <div class=md-source__repository> CIDRI-Africa/microbial-genomics-training </div> </a> </div> </nav> <nav class=md-tabs aria-label=Tabs data-md-component=tabs> <div class=md-grid> <ul class=md-tabs__list> <li class=md-tabs__item> <a href=../.. class=md-tabs__link> Home </a> </li> <li class=md-tabs__item> <a href=../../course/overview/ class=md-tabs__link> Course Information </a> </li> <li class="md-tabs__item md-tabs__item--active"> <a href=../day1/ class=md-tabs__link> Daily Modules </a> </li> <li class=md-tabs__item> <a href=../../datasets/ class=md-tabs__link> Resources </a> </li> </ul> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=../.. title="Microbial Genomics & Metagenomics Training Course" class="md-nav__button md-logo" aria-label="Microbial Genomics & Metagenomics Training Course" data-md-component=logo> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 2h2v2c0 1.44.68 2.61 1.88 3.78.86.83 2.01 1.63 3.21 2.42l-1.83 1.19C8.27 10.72 7.31 10 6.5 9.21 5.07 7.82 4 6.1 4 4zm14 0h2v2c0 2.1-1.07 3.82-2.5 5.21-1.41 1.38-3.21 2.52-4.96 3.63-1.75 1.12-3.45 2.21-4.66 3.38C6.68 17.39 6 18.56 6 20v2H4v-2c0-2.1 1.07-3.82 2.5-5.21 1.41-1.38 3.21-2.52 4.96-3.63 1.75-1.12 3.45-2.21 4.66-3.38C17.32 6.61 18 5.44 18 4zm-3.26 10.61c.99.67 1.95 1.39 2.76 2.18C18.93 16.18 20 17.9 20 20v2h-2v-2c0-1.44-.68-2.61-1.88-3.78-.86-.83-2.01-1.63-3.21-2.42zM7 3h10v1l-.06.5H7.06L7 4zm.68 3h8.64c-.24.34-.52.69-.9 1.06l-.51.44H9.07l-.49-.44c-.38-.37-.66-.72-.9-1.06m1.41 10.5h5.84l.49.44c.38.37.66.72.9 1.06H7.68c.24-.34.52-.69.9-1.06zm-2.03 3h9.88l.06.5v1H7v-1z"/></svg> </a> Microbial Genomics & Metagenomics Training Course </label> <div class=md-nav__source> <a href=https://github.com/CIDRI-Africa/microbial-genomics-training title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill=currentColor d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg> </div> <div class=md-source__repository> CIDRI-Africa/microbial-genomics-training </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. class=md-nav__link> <span class=md-ellipsis> Home </span> </a> </li> <li class="md-nav__item md-nav__item--pruned md-nav__item--nested"> <a href=../../course/overview/ class=md-nav__link> <span class=md-ellipsis> Course Information </span> <span class="md-nav__icon md-icon"></span> </a> </li> <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested"> <input class="md-nav__toggle md-toggle " type=checkbox id=__nav_3 checked> <div class="md-nav__link md-nav__container"> <a href=../day1/ class="md-nav__link "> <span class=md-ellipsis> Daily Modules </span> </a> <label class="md-nav__link " for=__nav_3 id=__nav_3_label tabindex> <span class="md-nav__icon md-icon"></span> </label> </div> <nav class=md-nav data-md-level=1 aria-labelledby=__nav_3_label aria-expanded=true> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Daily Modules </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../day2/ class=md-nav__link> <span class=md-ellipsis> Day 2 - Introduction to Commandline, High Performance Computing, & Quality Control </span> </a> </li> <li class=md-nav__item> <a href=../day3/ class=md-nav__link> <span class=md-ellipsis> Day 3 - Genomic Characterization </span> </a> </li> <li class=md-nav__item> <a href=../day4/ class=md-nav__link> <span class=md-ellipsis> Day 4 - Genomic Characterization </span> </a> </li> <li class=md-nav__item> <a href=../day5/ class=md-nav__link> <span class=md-ellipsis> Day 5 - Nextflow </span> </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> <span class=md-ellipsis> Day 6 - Nextflow Pipeline Development </span> <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> <span class=md-ellipsis> Day 6 - Nextflow Pipeline Development </span> </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#overview class=md-nav__link> <span class=md-ellipsis> Overview </span> </a> </li> <li class=md-nav__item> <a href=#learning-objectives class=md-nav__link> <span class=md-ellipsis> Learning Objectives </span> </a> </li> <li class=md-nav__item> <a href=#schedule class=md-nav__link> <span class=md-ellipsis> Schedule </span> </a> </li> <li class=md-nav__item> <a href=#key-topics class=md-nav__link> <span class=md-ellipsis> Key Topics </span> </a> <nav class=md-nav aria-label="Key Topics"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#1-rare-pathogen-detection class=md-nav__link> <span class=md-ellipsis> 1. Rare Pathogen Detection </span> </a> </li> <li class=md-nav__item> <a href=#2-co-infection-analysis class=md-nav__link> <span class=md-ellipsis> 2. Co-infection Analysis </span> </a> </li> <li class=md-nav__item> <a href=#3-community-dynamics class=md-nav__link> <span class=md-ellipsis> 3. Community Dynamics </span> </a> </li> <li class=md-nav__item> <a href=#4-variant-calling class=md-nav__link> <span class=md-ellipsis> 4. Variant Calling </span> </a> </li> <li class=md-nav__item> <a href=#5-genotype-phenotype-correlation class=md-nav__link> <span class=md-ellipsis> 5. Genotype-Phenotype Correlation </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#tools-and-methods class=md-nav__link> <span class=md-ellipsis> Tools and Methods </span> </a> <nav class=md-nav aria-label="Tools and Methods"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#pathogen-detection class=md-nav__link> <span class=md-ellipsis> Pathogen Detection </span> </a> </li> <li class=md-nav__item> <a href=#variant-calling-tools class=md-nav__link> <span class=md-ellipsis> Variant Calling Tools </span> </a> </li> <li class=md-nav__item> <a href=#co-infection-analysis class=md-nav__link> <span class=md-ellipsis> Co-infection Analysis </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#hands-on-exercises class=md-nav__link> <span class=md-ellipsis> Hands-on Exercises </span> </a> <nav class=md-nav aria-label="Hands-on Exercises"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#exercise-1-rare-pathogen-detection-60-minutes class=md-nav__link> <span class=md-ellipsis> Exercise 1: Rare Pathogen Detection (60 minutes) </span> </a> </li> <li class=md-nav__item> <a href=#exercise-2-co-infection-analysis-75-minutes class=md-nav__link> <span class=md-ellipsis> Exercise 2: Co-infection Analysis (75 minutes) </span> </a> </li> <li class=md-nav__item> <a href=#exercise-3-variant-calling-90-minutes class=md-nav__link> <span class=md-ellipsis> Exercise 3: Variant Calling (90 minutes) </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#key-concepts class=md-nav__link> <span class=md-ellipsis> Key Concepts </span> </a> <nav class=md-nav aria-label="Key Concepts"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#detection-sensitivity class=md-nav__link> <span class=md-ellipsis> Detection Sensitivity </span> </a> </li> <li class=md-nav__item> <a href=#co-infection-dynamics class=md-nav__link> <span class=md-ellipsis> Co-infection Dynamics </span> </a> </li> <li class=md-nav__item> <a href=#variant-classification class=md-nav__link> <span class=md-ellipsis> Variant Classification </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#clinical-applications class=md-nav__link> <span class=md-ellipsis> Clinical Applications </span> </a> <nav class=md-nav aria-label="Clinical Applications"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#diagnostic-applications class=md-nav__link> <span class=md-ellipsis> Diagnostic Applications </span> </a> </li> <li class=md-nav__item> <a href=#antimicrobial-resistance class=md-nav__link> <span class=md-ellipsis> Antimicrobial Resistance </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#assessment-activities class=md-nav__link> <span class=md-ellipsis> Assessment Activities </span> </a> <nav class=md-nav aria-label="Assessment Activities"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#individual-analysis class=md-nav__link> <span class=md-ellipsis> Individual Analysis </span> </a> </li> <li class=md-nav__item> <a href=#group-discussion class=md-nav__link> <span class=md-ellipsis> Group Discussion </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#common-challenges class=md-nav__link> <span class=md-ellipsis> Common Challenges </span> </a> <nav class=md-nav aria-label="Common Challenges"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#false-positive-control class=md-nav__link> <span class=md-ellipsis> False Positive Control </span> </a> </li> <li class=md-nav__item> <a href=#low-coverage-variants class=md-nav__link> <span class=md-ellipsis> Low Coverage Variants </span> </a> </li> <li class=md-nav__item> <a href=#complex-structural-variants class=md-nav__link> <span class=md-ellipsis> Complex Structural Variants </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#resources class=md-nav__link> <span class=md-ellipsis> Resources </span> </a> <nav class=md-nav aria-label=Resources> <ul class=md-nav__list> <li class=md-nav__item> <a href=#essential-reading class=md-nav__link> <span class=md-ellipsis> Essential Reading </span> </a> </li> <li class=md-nav__item> <a href=#software-documentation class=md-nav__link> <span class=md-ellipsis> Software Documentation </span> </a> </li> <li class=md-nav__item> <a href=#databases class=md-nav__link> <span class=md-ellipsis> Databases </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#looking-ahead class=md-nav__link> <span class=md-ellipsis> Looking Ahead </span> </a> </li> <li class=md-nav__item> <a href=#homework-optional class=md-nav__link> <span class=md-ellipsis> Homework (Optional) </span> </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../day7/ class=md-nav__link> <span class=md-ellipsis> Day 7 - Metagenomic profiling </span> </a> </li> <li class=md-nav__item> <a href=../day8/ class=md-nav__link> <span class=md-ellipsis> Day 8 - Comparative Genomics </span> </a> </li> <li class=md-nav__item> <a href=../day9/ class=md-nav__link> <span class=md-ellipsis> Day 9 - Bring your own data </span> </a> </li> <li class=md-nav__item> <a href=../day10/ class=md-nav__link> <span class=md-ellipsis> Day 10 - Wrap-up session </span> </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--pruned md-nav__item--nested"> <a href=../../datasets/ class=md-nav__link> <span class=md-ellipsis> Resources </span> <span class="md-nav__icon md-icon"></span> </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/CIDRI-Africa/microbial-genomics-training/edit/master/docs/modules/day6/index.md title="Edit this page" class="md-content__button md-icon" rel=edit> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4zm10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1z"/></svg> </a> <a href=https://github.com/CIDRI-Africa/microbial-genomics-training/raw/master/docs/modules/day6/index.md title="View source of this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M17 18c.56 0 1 .44 1 1s-.44 1-1 1-1-.44-1-1 .44-1 1-1m0-3c-2.73 0-5.06 1.66-6 4 .94 2.34 3.27 4 6 4s5.06-1.66 6-4c-.94-2.34-3.27-4-6-4m0 6.5a2.5 2.5 0 0 1-2.5-2.5 2.5 2.5 0 0 1 2.5-2.5 2.5 2.5 0 0 1 2.5 2.5 2.5 2.5 0 0 1-2.5 2.5M9.27 20H6V4h7v5h5v4.07c.7.08 1.36.25 2 .49V8l-6-6H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h4.5a8.2 8.2 0 0 1-1.23-2"/></svg> </a> <h1 id=day-6-nextflow-pipeline-development>Day 6: Nextflow Pipeline Development<a class=headerlink href=#day-6-nextflow-pipeline-development title="Permanent link">&para;</a></h1> <p><strong>Date</strong>: September 8, 2025<br> <strong>Duration</strong>: 09:00-13:00 CAT<br> <strong>Focus</strong>: Continued pipeline development, testing, and execution</p> <h2 id=overview>Overview<a class=headerlink href=#overview title="Permanent link">&para;</a></h2> <p>Day 6 continues from Day 5 with hands-on Nextflow pipeline development. Participants will complete their genomic analysis pipelines, learn advanced Nextflow features, implement testing strategies, and execute their pipelines on real data. This practical session reinforces workflow development skills through implementation.</p> <h2 id=learning-objectives>Learning Objectives<a class=headerlink href=#learning-objectives title="Permanent link">&para;</a></h2> <p>By the end of Day 6, you will be able to:</p> <ul> <li>Complete a functional Nextflow pipeline for genomic analysis</li> <li>Implement advanced Nextflow features (error handling, checkpoints, caching)</li> <li>Test and debug Nextflow pipelines effectively</li> <li>Optimize pipeline performance and resource usage</li> <li>Execute pipelines on HPC systems using different executors</li> <li>Monitor and troubleshoot pipeline runs</li> </ul> <h2 id=schedule>Schedule<a class=headerlink href=#schedule title="Permanent link">&para;</a></h2> <table> <thead> <tr> <th>Time (CAT)</th> <th>Topic</th> <th>Links</th> <th>Trainer</th> </tr> </thead> <tbody> <tr> <td><strong>09:00</strong></td> <td><em>Pipeline development continued and running</em></td> <td></td> <td>Mamana Mbiyavanga</td> </tr> <tr> <td><strong>11:30</strong></td> <td><strong>Break</strong></td> <td></td> <td></td> </tr> <tr> <td><strong>12:00</strong></td> <td><em>Pipeline development continued and running</em></td> <td></td> <td>Mamana Mbiyavanga</td> </tr> </tbody> </table> <h2 id=key-topics>Key Topics<a class=headerlink href=#key-topics title="Permanent link">&para;</a></h2> <h3 id=1-rare-pathogen-detection>1. Rare Pathogen Detection<a class=headerlink href=#1-rare-pathogen-detection title="Permanent link">&para;</a></h3> <ul> <li>Sensitivity thresholds for pathogen detection</li> <li>Database selection for comprehensive pathogen identification</li> <li>Distinguishing true positives from contamination</li> <li>Novel pathogen discovery approaches</li> </ul> <h3 id=2-co-infection-analysis>2. Co-infection Analysis<a class=headerlink href=#2-co-infection-analysis title="Permanent link">&para;</a></h3> <ul> <li>Multi-pathogen detection in single samples</li> <li>Strain-level resolution in mixed infections</li> <li>Quantitative analysis of pathogen loads</li> <li>Clinical significance of co-infections</li> </ul> <h3 id=3-community-dynamics>3. Community Dynamics<a class=headerlink href=#3-community-dynamics title="Permanent link">&para;</a></h3> <ul> <li>Temporal changes in microbiome composition</li> <li>Treatment-induced community shifts</li> <li>Ecological succession patterns</li> <li>Stability and resilience metrics</li> </ul> <h3 id=4-variant-calling>4. Variant Calling<a class=headerlink href=#4-variant-calling title="Permanent link">&para;</a></h3> <ul> <li>SNP and indel detection from genomic data</li> <li>Quality filtering and validation of variants</li> <li>Structural variant identification</li> <li>Population genetics analysis</li> </ul> <h3 id=5-genotype-phenotype-correlation>5. Genotype-Phenotype Correlation<a class=headerlink href=#5-genotype-phenotype-correlation title="Permanent link">&para;</a></h3> <ul> <li>Resistance gene identification and validation</li> <li>Phenotypic testing correlation with genomic predictions</li> <li>Machine learning approaches for phenotype prediction</li> <li>Clinical decision support applications</li> </ul> <h2 id=tools-and-methods>Tools and Methods<a class=headerlink href=#tools-and-methods title="Permanent link">&para;</a></h2> <h3 id=pathogen-detection>Pathogen Detection<a class=headerlink href=#pathogen-detection title="Permanent link">&para;</a></h3> <ul> <li><strong>PathSeq</strong> - Pathogen detection in clinical samples</li> <li><strong>Kraken2</strong> with comprehensive databases</li> <li><strong>MetaPhlAn4</strong> for species-level identification</li> <li><strong>BLAST</strong> for targeted sequence search</li> </ul> <h3 id=variant-calling-tools>Variant Calling Tools<a class=headerlink href=#variant-calling-tools title="Permanent link">&para;</a></h3> <ul> <li><strong>GATK</strong> - Genome Analysis Toolkit</li> <li><strong>FreeBayes</strong> - Bayesian variant caller</li> <li><strong>VarScan</strong> - Somatic and germline variant calling</li> <li><strong>SnpEff</strong> - Variant annotation and effect prediction</li> </ul> <h3 id=co-infection-analysis>Co-infection Analysis<a class=headerlink href=#co-infection-analysis title="Permanent link">&para;</a></h3> <ul> <li><strong>StrainEst</strong> - Strain-level analysis</li> <li><strong>ConStrains</strong> - Strain reconstruction</li> <li><strong>DESMAN</strong> - De novo extraction of strains from metagenomes</li> </ul> <h2 id=hands-on-exercises>Hands-on Exercises<a class=headerlink href=#hands-on-exercises title="Permanent link">&para;</a></h2> <h3 id=exercise-1-rare-pathogen-detection-60-minutes>Exercise 1: Rare Pathogen Detection (60 minutes)<a class=headerlink href=#exercise-1-rare-pathogen-detection-60-minutes title="Permanent link">&para;</a></h3> <p>Use metagenomic data to identify rare pathogens in clinical samples.</p> <div class="language-bash highlight"><pre><span></span><code><span id=__span-0-1><a id=__codelineno-0-1 name=__codelineno-0-1 href=#__codelineno-0-1></a><span class=c1># Run Kraken2 with comprehensive database</span>
</span><span id=__span-0-2><a id=__codelineno-0-2 name=__codelineno-0-2 href=#__codelineno-0-2></a>kraken2<span class=w> </span>--db<span class=w> </span>kraken2_plus<span class=w> </span>--paired<span class=w> </span>sample_R1.fastq<span class=w> </span>sample_R2.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-0-3><a id=__codelineno-0-3 name=__codelineno-0-3 href=#__codelineno-0-3></a><span class=w>    </span>--output<span class=w> </span>sample.kraken<span class=w> </span>--report<span class=w> </span>sample_report.txt
</span><span id=__span-0-4><a id=__codelineno-0-4 name=__codelineno-0-4 href=#__codelineno-0-4></a>
</span><span id=__span-0-5><a id=__codelineno-0-5 name=__codelineno-0-5 href=#__codelineno-0-5></a><span class=c1># Filter for potential pathogens</span>
</span><span id=__span-0-6><a id=__codelineno-0-6 name=__codelineno-0-6 href=#__codelineno-0-6></a>python3<span class=w> </span>scripts/filter_pathogens.py<span class=w> </span>sample_report.txt<span class=w> </span><span class=se>\</span>
</span><span id=__span-0-7><a id=__codelineno-0-7 name=__codelineno-0-7 href=#__codelineno-0-7></a><span class=w>    </span>--pathogen-list<span class=w> </span>pathogen_database.txt<span class=w> </span><span class=se>\</span>
</span><span id=__span-0-8><a id=__codelineno-0-8 name=__codelineno-0-8 href=#__codelineno-0-8></a><span class=w>    </span>--min-reads<span class=w> </span><span class=m>10</span><span class=w> </span>--output<span class=w> </span>rare_pathogens.txt
</span><span id=__span-0-9><a id=__codelineno-0-9 name=__codelineno-0-9 href=#__codelineno-0-9></a>
</span><span id=__span-0-10><a id=__codelineno-0-10 name=__codelineno-0-10 href=#__codelineno-0-10></a><span class=c1># Validate findings with targeted BLAST</span>
</span><span id=__span-0-11><a id=__codelineno-0-11 name=__codelineno-0-11 href=#__codelineno-0-11></a>blastn<span class=w> </span>-query<span class=w> </span>extracted_sequences.fasta<span class=w> </span>-db<span class=w> </span>nt<span class=w> </span><span class=se>\</span>
</span><span id=__span-0-12><a id=__codelineno-0-12 name=__codelineno-0-12 href=#__codelineno-0-12></a><span class=w>    </span>-outfmt<span class=w> </span><span class=s2>&quot;6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle&quot;</span><span class=w> </span><span class=se>\</span>
</span><span id=__span-0-13><a id=__codelineno-0-13 name=__codelineno-0-13 href=#__codelineno-0-13></a><span class=w>    </span>-max_target_seqs<span class=w> </span><span class=m>5</span><span class=w> </span>-out<span class=w> </span>validation_blast.txt
</span></code></pre></div> <h3 id=exercise-2-co-infection-analysis-75-minutes>Exercise 2: Co-infection Analysis (75 minutes)<a class=headerlink href=#exercise-2-co-infection-analysis-75-minutes title="Permanent link">&para;</a></h3> <p>Analyze samples with multiple pathogen species.</p> <div class="language-bash highlight"><pre><span></span><code><span id=__span-1-1><a id=__codelineno-1-1 name=__codelineno-1-1 href=#__codelineno-1-1></a><span class=c1># Quantify pathogen abundances</span>
</span><span id=__span-1-2><a id=__codelineno-1-2 name=__codelineno-1-2 href=#__codelineno-1-2></a>metaphlan<span class=w> </span>sample_metagenome.fastq.gz<span class=w> </span>--input_type<span class=w> </span>fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-1-3><a id=__codelineno-1-3 name=__codelineno-1-3 href=#__codelineno-1-3></a><span class=w>    </span>--bowtie2out<span class=w> </span>sample_bowtie2.bz2<span class=w> </span>--nproc<span class=w> </span><span class=m>4</span><span class=w> </span><span class=se>\</span>
</span><span id=__span-1-4><a id=__codelineno-1-4 name=__codelineno-1-4 href=#__codelineno-1-4></a><span class=w>    </span>--output_file<span class=w> </span>sample_profile.txt
</span><span id=__span-1-5><a id=__codelineno-1-5 name=__codelineno-1-5 href=#__codelineno-1-5></a>
</span><span id=__span-1-6><a id=__codelineno-1-6 name=__codelineno-1-6 href=#__codelineno-1-6></a><span class=c1># Identify co-infections</span>
</span><span id=__span-1-7><a id=__codelineno-1-7 name=__codelineno-1-7 href=#__codelineno-1-7></a>python3<span class=w> </span>scripts/coinfection_analysis.py<span class=w> </span>sample_profile.txt<span class=w> </span><span class=se>\</span>
</span><span id=__span-1-8><a id=__codelineno-1-8 name=__codelineno-1-8 href=#__codelineno-1-8></a><span class=w>    </span>--abundance-threshold<span class=w> </span><span class=m>0</span>.1<span class=w> </span>--output<span class=w> </span>coinfection_report.txt
</span><span id=__span-1-9><a id=__codelineno-1-9 name=__codelineno-1-9 href=#__codelineno-1-9></a>
</span><span id=__span-1-10><a id=__codelineno-1-10 name=__codelineno-1-10 href=#__codelineno-1-10></a><span class=c1># Strain-level analysis for dominant pathogens</span>
</span><span id=__span-1-11><a id=__codelineno-1-11 name=__codelineno-1-11 href=#__codelineno-1-11></a>strainest<span class=w> </span>sample_bowtie2.bz2<span class=w> </span>reference_genomes/<span class=w> </span><span class=se>\</span>
</span><span id=__span-1-12><a id=__codelineno-1-12 name=__codelineno-1-12 href=#__codelineno-1-12></a><span class=w>    </span>--output<span class=w> </span>strain_analysis/
</span></code></pre></div> <h3 id=exercise-3-variant-calling-90-minutes>Exercise 3: Variant Calling (90 minutes)<a class=headerlink href=#exercise-3-variant-calling-90-minutes title="Permanent link">&para;</a></h3> <p>Call variants and correlate with phenotypic data.</p> <div class="language-bash highlight"><pre><span></span><code><span id=__span-2-1><a id=__codelineno-2-1 name=__codelineno-2-1 href=#__codelineno-2-1></a><span class=c1># Map reads to reference genome</span>
</span><span id=__span-2-2><a id=__codelineno-2-2 name=__codelineno-2-2 href=#__codelineno-2-2></a>bwa<span class=w> </span>mem<span class=w> </span>reference.fasta<span class=w> </span>sample_R1.fastq<span class=w> </span>sample_R2.fastq<span class=w> </span><span class=p>|</span><span class=w> </span><span class=se>\</span>
</span><span id=__span-2-3><a id=__codelineno-2-3 name=__codelineno-2-3 href=#__codelineno-2-3></a><span class=w>    </span>samtools<span class=w> </span>sort<span class=w> </span>-o<span class=w> </span>sample_sorted.bam
</span><span id=__span-2-4><a id=__codelineno-2-4 name=__codelineno-2-4 href=#__codelineno-2-4></a>samtools<span class=w> </span>index<span class=w> </span>sample_sorted.bam
</span><span id=__span-2-5><a id=__codelineno-2-5 name=__codelineno-2-5 href=#__codelineno-2-5></a>
</span><span id=__span-2-6><a id=__codelineno-2-6 name=__codelineno-2-6 href=#__codelineno-2-6></a><span class=c1># Call variants with GATK</span>
</span><span id=__span-2-7><a id=__codelineno-2-7 name=__codelineno-2-7 href=#__codelineno-2-7></a>gatk<span class=w> </span>HaplotypeCaller<span class=w> </span>-R<span class=w> </span>reference.fasta<span class=w> </span>-I<span class=w> </span>sample_sorted.bam<span class=w> </span><span class=se>\</span>
</span><span id=__span-2-8><a id=__codelineno-2-8 name=__codelineno-2-8 href=#__codelineno-2-8></a><span class=w>    </span>-O<span class=w> </span>sample_variants.vcf
</span><span id=__span-2-9><a id=__codelineno-2-9 name=__codelineno-2-9 href=#__codelineno-2-9></a>
</span><span id=__span-2-10><a id=__codelineno-2-10 name=__codelineno-2-10 href=#__codelineno-2-10></a><span class=c1># Annotate variants</span>
</span><span id=__span-2-11><a id=__codelineno-2-11 name=__codelineno-2-11 href=#__codelineno-2-11></a>snpEff<span class=w> </span>ann<span class=w> </span>reference_genome<span class=w> </span>sample_variants.vcf<span class=w> </span>&gt;<span class=w> </span>sample_annotated.vcf
</span><span id=__span-2-12><a id=__codelineno-2-12 name=__codelineno-2-12 href=#__codelineno-2-12></a>
</span><span id=__span-2-13><a id=__codelineno-2-13 name=__codelineno-2-13 href=#__codelineno-2-13></a><span class=c1># Correlate with phenotype data</span>
</span><span id=__span-2-14><a id=__codelineno-2-14 name=__codelineno-2-14 href=#__codelineno-2-14></a>python3<span class=w> </span>scripts/genotype_phenotype_correlation.py<span class=w> </span><span class=se>\</span>
</span><span id=__span-2-15><a id=__codelineno-2-15 name=__codelineno-2-15 href=#__codelineno-2-15></a><span class=w>    </span>sample_annotated.vcf<span class=w> </span>phenotype_data.csv<span class=w> </span><span class=se>\</span>
</span><span id=__span-2-16><a id=__codelineno-2-16 name=__codelineno-2-16 href=#__codelineno-2-16></a><span class=w>    </span>--output<span class=w> </span>correlation_analysis.txt
</span></code></pre></div> <h2 id=key-concepts>Key Concepts<a class=headerlink href=#key-concepts title="Permanent link">&para;</a></h2> <h3 id=detection-sensitivity>Detection Sensitivity<a class=headerlink href=#detection-sensitivity title="Permanent link">&para;</a></h3> <ul> <li><strong>Limit of detection</strong>: Minimum number of reads for reliable identification</li> <li><strong>False positive control</strong>: Contamination and database artifacts</li> <li><strong>Validation requirements</strong>: Confirmatory testing approaches</li> <li><strong>Clinical significance</strong>: Distinguishing colonization from infection</li> </ul> <h3 id=co-infection-dynamics>Co-infection Dynamics<a class=headerlink href=#co-infection-dynamics title="Permanent link">&para;</a></h3> <ul> <li><strong>Competitive exclusion</strong>: Pathogens competing for resources</li> <li><strong>Synergistic interactions</strong>: Enhanced pathogenicity in co-infections</li> <li><strong>Treatment complications</strong>: Multi-drug considerations</li> <li><strong>Epidemiological implications</strong>: Transmission patterns</li> </ul> <h3 id=variant-classification>Variant Classification<a class=headerlink href=#variant-classification title="Permanent link">&para;</a></h3> <table> <thead> <tr> <th>Variant Type</th> <th>Impact</th> <th>Example</th> </tr> </thead> <tbody> <tr> <td>Synonymous</td> <td>Silent (no amino acid change)</td> <td>c.123C&gt;T (p.Leu41=)</td> </tr> <tr> <td>Missense</td> <td>Amino acid change</td> <td>c.123G&gt;A (p.Gly41Ser)</td> </tr> <tr> <td>Nonsense</td> <td>Premature stop codon</td> <td>c.123C&gt;T (p.Gln41*)</td> </tr> <tr> <td>Frameshift</td> <td>Reading frame alteration</td> <td>c.123delA (p.Leu41fs)</td> </tr> </tbody> </table> <h2 id=clinical-applications>Clinical Applications<a class=headerlink href=#clinical-applications title="Permanent link">&para;</a></h2> <h3 id=diagnostic-applications>Diagnostic Applications<a class=headerlink href=#diagnostic-applications title="Permanent link">&para;</a></h3> <ul> <li><strong>Syndromic testing</strong>: Multiple pathogen panels</li> <li><strong>Outbreak investigation</strong>: Source identification</li> <li><strong>Treatment monitoring</strong>: Resistance development</li> <li><strong>Infection control</strong>: Transmission prevention</li> </ul> <h3 id=antimicrobial-resistance>Antimicrobial Resistance<a class=headerlink href=#antimicrobial-resistance title="Permanent link">&para;</a></h3> <ul> <li><strong>Rapid resistance testing</strong>: Genomic predictions</li> <li><strong>Resistance mechanism identification</strong>: Gene and mutation analysis</li> <li><strong>Treatment guidance</strong>: Targeted therapy selection</li> <li><strong>Surveillance</strong>: Population-level resistance monitoring</li> </ul> <h2 id=assessment-activities>Assessment Activities<a class=headerlink href=#assessment-activities title="Permanent link">&para;</a></h2> <h3 id=individual-analysis>Individual Analysis<a class=headerlink href=#individual-analysis title="Permanent link">&para;</a></h3> <ul> <li>Identify rare pathogens in provided metagenomic dataset</li> <li>Quantify co-infection burden in clinical samples</li> <li>Perform variant calling and annotation on genomic data</li> <li>Correlate genomic findings with resistance phenotypes</li> </ul> <h3 id=group-discussion>Group Discussion<a class=headerlink href=#group-discussion title="Permanent link">&para;</a></h3> <ul> <li>Compare sensitivity of different pathogen detection methods</li> <li>Interpret co-infection patterns in clinical context</li> <li>Evaluate clinical significance of identified variants</li> <li>Discuss challenges in genotype-phenotype correlation</li> </ul> <h2 id=common-challenges>Common Challenges<a class=headerlink href=#common-challenges title="Permanent link">&para;</a></h2> <h3 id=false-positive-control>False Positive Control<a class=headerlink href=#false-positive-control title="Permanent link">&para;</a></h3> <div class="language-bash highlight"><pre><span></span><code><span id=__span-3-1><a id=__codelineno-3-1 name=__codelineno-3-1 href=#__codelineno-3-1></a><span class=c1># Include negative controls in analysis</span>
</span><span id=__span-3-2><a id=__codelineno-3-2 name=__codelineno-3-2 href=#__codelineno-3-2></a>kraken2<span class=w> </span>--db<span class=w> </span>comprehensive_db<span class=w> </span>--paired<span class=w> </span><span class=se>\</span>
</span><span id=__span-3-3><a id=__codelineno-3-3 name=__codelineno-3-3 href=#__codelineno-3-3></a><span class=w>    </span>negative_control_R1.fastq<span class=w> </span>negative_control_R2.fastq<span class=w> </span><span class=se>\</span>
</span><span id=__span-3-4><a id=__codelineno-3-4 name=__codelineno-3-4 href=#__codelineno-3-4></a><span class=w>    </span>--output<span class=w> </span>neg_control.kraken<span class=w> </span>--report<span class=w> </span>neg_control_report.txt
</span><span id=__span-3-5><a id=__codelineno-3-5 name=__codelineno-3-5 href=#__codelineno-3-5></a>
</span><span id=__span-3-6><a id=__codelineno-3-6 name=__codelineno-3-6 href=#__codelineno-3-6></a><span class=c1># Remove contaminants found in negative controls</span>
</span><span id=__span-3-7><a id=__codelineno-3-7 name=__codelineno-3-7 href=#__codelineno-3-7></a>python3<span class=w> </span>scripts/remove_contaminants.py<span class=w> </span>sample_report.txt<span class=w> </span><span class=se>\</span>
</span><span id=__span-3-8><a id=__codelineno-3-8 name=__codelineno-3-8 href=#__codelineno-3-8></a><span class=w>    </span>neg_control_report.txt<span class=w> </span>--output<span class=w> </span>cleaned_sample_report.txt
</span></code></pre></div> <h3 id=low-coverage-variants>Low Coverage Variants<a class=headerlink href=#low-coverage-variants title="Permanent link">&para;</a></h3> <div class="language-bash highlight"><pre><span></span><code><span id=__span-4-1><a id=__codelineno-4-1 name=__codelineno-4-1 href=#__codelineno-4-1></a><span class=c1># Adjust variant calling parameters for low coverage</span>
</span><span id=__span-4-2><a id=__codelineno-4-2 name=__codelineno-4-2 href=#__codelineno-4-2></a>freebayes<span class=w> </span>-f<span class=w> </span>reference.fasta<span class=w> </span>--min-coverage<span class=w> </span><span class=m>5</span><span class=w> </span>--min-alternate-count<span class=w> </span><span class=m>2</span><span class=w> </span><span class=se>\</span>
</span><span id=__span-4-3><a id=__codelineno-4-3 name=__codelineno-4-3 href=#__codelineno-4-3></a><span class=w>    </span>--min-alternate-fraction<span class=w> </span><span class=m>0</span>.1<span class=w> </span>sample_sorted.bam<span class=w> </span>&gt;<span class=w> </span>low_cov_variants.vcf
</span><span id=__span-4-4><a id=__codelineno-4-4 name=__codelineno-4-4 href=#__codelineno-4-4></a>
</span><span id=__span-4-5><a id=__codelineno-4-5 name=__codelineno-4-5 href=#__codelineno-4-5></a><span class=c1># Validate with manual inspection</span>
</span><span id=__span-4-6><a id=__codelineno-4-6 name=__codelineno-4-6 href=#__codelineno-4-6></a>samtools<span class=w> </span>tview<span class=w> </span>sample_sorted.bam<span class=w> </span>reference.fasta<span class=w> </span>-p<span class=w> </span>chr1:12345
</span></code></pre></div> <h3 id=complex-structural-variants>Complex Structural Variants<a class=headerlink href=#complex-structural-variants title="Permanent link">&para;</a></h3> <div class="language-bash highlight"><pre><span></span><code><span id=__span-5-1><a id=__codelineno-5-1 name=__codelineno-5-1 href=#__codelineno-5-1></a><span class=c1># Use specialized tools for structural variants</span>
</span><span id=__span-5-2><a id=__codelineno-5-2 name=__codelineno-5-2 href=#__codelineno-5-2></a>delly<span class=w> </span>call<span class=w> </span>-g<span class=w> </span>reference.fasta<span class=w> </span>sample_sorted.bam<span class=w> </span>-o<span class=w> </span>structural_variants.bcf
</span><span id=__span-5-3><a id=__codelineno-5-3 name=__codelineno-5-3 href=#__codelineno-5-3></a>
</span><span id=__span-5-4><a id=__codelineno-5-4 name=__codelineno-5-4 href=#__codelineno-5-4></a><span class=c1># Convert and filter</span>
</span><span id=__span-5-5><a id=__codelineno-5-5 name=__codelineno-5-5 href=#__codelineno-5-5></a>bcftools<span class=w> </span>view<span class=w> </span>structural_variants.bcf<span class=w> </span><span class=p>|</span><span class=w> </span><span class=se>\</span>
</span><span id=__span-5-6><a id=__codelineno-5-6 name=__codelineno-5-6 href=#__codelineno-5-6></a><span class=w>    </span>bcftools<span class=w> </span>filter<span class=w> </span>-i<span class=w> </span><span class=s1>&#39;QUAL&gt;20&#39;</span><span class=w> </span>&gt;<span class=w> </span>filtered_sv.vcf
</span></code></pre></div> <h2 id=resources>Resources<a class=headerlink href=#resources title="Permanent link">&para;</a></h2> <h3 id=essential-reading>Essential Reading<a class=headerlink href=#essential-reading title="Permanent link">&para;</a></h3> <ul> <li>Simner et al. (2018). Understanding the promises and hurdles of metagenomic next-generation sequencing</li> <li>Chiu &amp; Miller (2019). Clinical metagenomics</li> <li>Ellington et al. (2017). The role of whole genome sequencing in antimicrobial susceptibility testing</li> </ul> <h3 id=software-documentation>Software Documentation<a class=headerlink href=#software-documentation title="Permanent link">&para;</a></h3> <ul> <li><a href=https://gatk.broadinstitute.org/hc/en-us/sections/360007226651>GATK Best Practices</a></li> <li><a href=https://gatk.broadinstitute.org/hc/en-us/articles/360037224932>PathSeq Documentation</a></li> <li><a href=https://pcingola.github.io/SnpEff/ >SnpEff Manual</a></li> </ul> <h3 id=databases>Databases<a class=headerlink href=#databases title="Permanent link">&para;</a></h3> <ul> <li><a href=https://card.mcmaster.ca/ >CARD</a> - Comprehensive Antibiotic Resistance Database</li> <li><a href=https://cge.cbs.dtu.dk/services/ResFinder/ >ResFinder</a> - Resistance gene identification</li> <li><a href=https://www.ncbi.nlm.nih.gov/pathogens/ >NCBI Pathogen Detection</a> - Pathogen surveillance</li> </ul> <h2 id=looking-ahead>Looking Ahead<a class=headerlink href=#looking-ahead title="Permanent link">&para;</a></h2> <p><strong>Day 7 Preview</strong>: Nextflow workflows including: - Introduction to reproducible workflow systems - Nextflow basics and nf-core community - Building automated analysis pipelines - Pipeline development best practices</p> <h2 id=homework-optional>Homework (Optional)<a class=headerlink href=#homework-optional title="Permanent link">&para;</a></h2> <ol> <li>Analyze co-infection patterns in additional clinical datasets</li> <li>Practice variant calling with different parameter settings</li> <li>Explore resistance gene databases for target pathogens</li> <li>Read case studies of metagenomic pathogen detection</li> </ol> <hr> <p><strong>Key Learning Outcome</strong>: Advanced metagenomic analysis enables detection of complex infection scenarios and genomic variants, providing critical information for clinical decision-making and public health surveillance.</p> </article> </div> <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script> <script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script> </div> <button type=button class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg> Back to top </button> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../day5/ class="md-footer__link md-footer__link--prev" aria-label="Previous: Day 5 - Nextflow"> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg> </div> <div class=md-footer__title> <span class=md-footer__direction> Previous </span> <div class=md-ellipsis> Day 5 - Nextflow </div> </div> </a> <a href=../day7/ class="md-footer__link md-footer__link--next" aria-label="Next: Day 7 - Metagenomic profiling"> <div class=md-footer__title> <span class=md-footer__direction> Next </span> <div class=md-ellipsis> Day 7 - Metagenomic profiling </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; 2024 CIDRI-Africa </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=https://github.com/CIDRI-Africa target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill=currentColor d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg> </a> <a href=https://twitter.com/CIDRIAfrica target=_blank rel=noopener title=twitter.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 512 512"><!-- Font Awesome Free 7.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path fill=currentColor d="M459.4 151.7c.3 4.5.3 9.1.3 13.6 0 138.7-105.6 298.6-298.6 298.6-59.5 0-114.7-17.2-161.1-47.1 8.4 1 16.6 1.3 25.3 1.3 49.1 0 94.2-16.6 130.3-44.8-46.1-1-84.8-31.2-98.1-72.8 6.5 1 13 1.6 19.8 1.6 9.4 0 18.8-1.3 27.6-3.6-48.1-9.7-84.1-52-84.1-103v-1.3c14 7.8 30.2 12.7 47.4 13.3-28.3-18.8-46.8-51-46.8-87.4 0-19.5 5.2-37.4 14.3-53C87.4 130.8 165 172.4 252.1 176.9c-1.6-7.8-2.6-15.9-2.6-24C249.5 95.1 296.3 48 354.4 48c30.2 0 57.5 12.7 76.7 33.1 23.7-4.5 46.5-13.3 66.6-25.3-7.8 24.4-24.4 44.8-46.1 57.8 21.1-2.3 41.6-8.1 60.4-16.2-14.3 20.8-32.2 39.3-52.6 54.3"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <div class=md-progress data-md-component=progress role=progressbar></div> <script id=__config type=application/json>{"base": "../..", "features": ["announce.dismiss", "content.action.edit", "content.action.view", "content.code.annotate", "content.code.copy", "content.code.select", "content.tabs.link", "content.tooltips", "header.autohide", "navigation.expand", "navigation.footer", "navigation.indexes", "navigation.instant", "navigation.instant.prefetch", "navigation.instant.progress", "navigation.prune", "navigation.sections", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow", "toc.integrate"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script> <script src=../../assets/javascripts/bundle.92b07e13.min.js></script> <script src=../../javascripts/extra.js></script> <script src=../../javascripts/mathjax.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> <script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(() => { lightbox.reload() });
</script></body> </html>